2024-02-26 18:25:05,436 - DEBUG - Starting new HTTPS connection (1): www.bitstamp.net:443
2024-02-26 18:25:06,778 - DEBUG - https://www.bitstamp.net:443 "GET /api/v2/ohlc/btcusd/?step=3600&limit=1 HTTP/1.1" 200 None
2024-02-26 18:25:11,741 - DEBUG - Starting new HTTP connection (1): 127.0.0.1:8000
2024-02-26 18:25:11,770 - DEBUG - http://127.0.0.1:8000 "POST /scrap/api/bitstampminute/add HTTP/1.1" 200 327
2024-02-26 18:25:59,237 - INFO - C:\Other Project\Ads\Bitcoin Scraping\scrapperE\scrap\views.py changed, reloading.
2024-02-26 18:26:31,208 - INFO - C:\Other Project\Ads\Bitcoin Scraping\scrapperE\scrap\views.py changed, reloading.
2024-02-26 18:27:27,950 - INFO - C:\Other Project\Ads\Bitcoin Scraping\scrapperE\scrap\views.py changed, reloading.
2024-02-26 18:27:54,890 - INFO - C:\Other Project\Ads\Bitcoin Scraping\scrapperE\scrap\views.py changed, reloading.
2024-02-26 18:28:02,650 - INFO - C:\Other Project\Ads\Bitcoin Scraping\scrapperE\scrap\views.py changed, reloading.
2024-02-26 18:28:35,554 - INFO - C:\Other Project\Ads\Bitcoin Scraping\scrapperE\scrap\views.py changed, reloading.
2024-02-26 18:30:05,133 - DEBUG - Starting new HTTPS connection (1): www.bitstamp.net:443
2024-02-26 18:30:27,404 - ERROR - Internal Server Error: /scrap/api/bitstamp/save_minutely
Traceback (most recent call last):
  File "C:\Users\Zhong_ming_Tan\AppData\Local\anaconda3\envs\fws-dell\Lib\site-packages\urllib3\connection.py", line 174, in _new_conn
    conn = connection.create_connection(
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Zhong_ming_Tan\AppData\Local\anaconda3\envs\fws-dell\Lib\site-packages\urllib3\util\connection.py", line 95, in create_connection
    raise err
  File "C:\Users\Zhong_ming_Tan\AppData\Local\anaconda3\envs\fws-dell\Lib\site-packages\urllib3\util\connection.py", line 85, in create_connection
    sock.connect(sa)
TimeoutError: [WinError 10060] A connection attempt failed because the connected party did not properly respond after a period of time, or established connection failed because connected host has failed to respond

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "C:\Users\Zhong_ming_Tan\AppData\Local\anaconda3\envs\fws-dell\Lib\site-packages\urllib3\connectionpool.py", line 715, in urlopen
    httplib_response = self._make_request(
                       ^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Zhong_ming_Tan\AppData\Local\anaconda3\envs\fws-dell\Lib\site-packages\urllib3\connectionpool.py", line 404, in _make_request
    self._validate_conn(conn)
  File "C:\Users\Zhong_ming_Tan\AppData\Local\anaconda3\envs\fws-dell\Lib\site-packages\urllib3\connectionpool.py", line 1058, in _validate_conn
    conn.connect()
  File "C:\Users\Zhong_ming_Tan\AppData\Local\anaconda3\envs\fws-dell\Lib\site-packages\urllib3\connection.py", line 363, in connect
    self.sock = conn = self._new_conn()
                       ^^^^^^^^^^^^^^^^
  File "C:\Users\Zhong_ming_Tan\AppData\Local\anaconda3\envs\fws-dell\Lib\site-packages\urllib3\connection.py", line 179, in _new_conn
    raise ConnectTimeoutError(
urllib3.exceptions.ConnectTimeoutError: (<urllib3.connection.HTTPSConnection object at 0x000001B2888CB080>, 'Connection to www.bitstamp.net timed out. (connect timeout=None)')

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "C:\Users\Zhong_ming_Tan\AppData\Local\anaconda3\envs\fws-dell\Lib\site-packages\requests\adapters.py", line 486, in send
    resp = conn.urlopen(
           ^^^^^^^^^^^^^
  File "C:\Users\Zhong_ming_Tan\AppData\Local\anaconda3\envs\fws-dell\Lib\site-packages\urllib3\connectionpool.py", line 799, in urlopen
    retries = retries.increment(
              ^^^^^^^^^^^^^^^^^^
  File "C:\Users\Zhong_ming_Tan\AppData\Local\anaconda3\envs\fws-dell\Lib\site-packages\urllib3\util\retry.py", line 592, in increment
    raise MaxRetryError(_pool, url, error or ResponseError(cause))
urllib3.exceptions.MaxRetryError: HTTPSConnectionPool(host='www.bitstamp.net', port=443): Max retries exceeded with url: /api/v2/ohlc/btcusd/?step=3600&limit=1 (Caused by ConnectTimeoutError(<urllib3.connection.HTTPSConnection object at 0x000001B2888CB080>, 'Connection to www.bitstamp.net timed out. (connect timeout=None)'))

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "C:\Other Project\Ads\Bitcoin Scraping\scrapperE\scrap\views.py", line 43, in call_bitsamp_api
    data = requests.get(ohlc_url)
           ^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Zhong_ming_Tan\AppData\Local\anaconda3\envs\fws-dell\Lib\site-packages\requests\api.py", line 73, in get
    return request("get", url, params=params, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Zhong_ming_Tan\AppData\Local\anaconda3\envs\fws-dell\Lib\site-packages\requests\api.py", line 59, in request
    return session.request(method=method, url=url, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Zhong_ming_Tan\AppData\Local\anaconda3\envs\fws-dell\Lib\site-packages\requests\sessions.py", line 589, in request
    resp = self.send(prep, **send_kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Zhong_ming_Tan\AppData\Local\anaconda3\envs\fws-dell\Lib\site-packages\requests\sessions.py", line 703, in send
    r = adapter.send(request, **kwargs)
        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Zhong_ming_Tan\AppData\Local\anaconda3\envs\fws-dell\Lib\site-packages\requests\adapters.py", line 507, in send
    raise ConnectTimeout(e, request=request)
requests.exceptions.ConnectTimeout: HTTPSConnectionPool(host='www.bitstamp.net', port=443): Max retries exceeded with url: /api/v2/ohlc/btcusd/?step=3600&limit=1 (Caused by ConnectTimeoutError(<urllib3.connection.HTTPSConnection object at 0x000001B2888CB080>, 'Connection to www.bitstamp.net timed out. (connect timeout=None)'))

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "C:\Users\Zhong_ming_Tan\AppData\Local\anaconda3\envs\fws-dell\Lib\site-packages\django\core\handlers\exception.py", line 55, in inner
    response = get_response(request)
               ^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Zhong_ming_Tan\AppData\Local\anaconda3\envs\fws-dell\Lib\site-packages\django\core\handlers\base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Zhong_ming_Tan\AppData\Local\anaconda3\envs\fws-dell\Lib\site-packages\django\views\decorators\csrf.py", line 65, in _view_wrapper
    return view_func(request, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Zhong_ming_Tan\AppData\Local\anaconda3\envs\fws-dell\Lib\site-packages\django\views\generic\base.py", line 104, in view
    return self.dispatch(request, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Zhong_ming_Tan\AppData\Local\anaconda3\envs\fws-dell\Lib\site-packages\rest_framework\views.py", line 509, in dispatch
    response = self.handle_exception(exc)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Zhong_ming_Tan\AppData\Local\anaconda3\envs\fws-dell\Lib\site-packages\rest_framework\views.py", line 469, in handle_exception
    self.raise_uncaught_exception(exc)
  File "C:\Users\Zhong_ming_Tan\AppData\Local\anaconda3\envs\fws-dell\Lib\site-packages\rest_framework\views.py", line 480, in raise_uncaught_exception
    raise exc
  File "C:\Users\Zhong_ming_Tan\AppData\Local\anaconda3\envs\fws-dell\Lib\site-packages\rest_framework\views.py", line 506, in dispatch
    response = handler(request, *args, **kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Zhong_ming_Tan\AppData\Local\anaconda3\envs\fws-dell\Lib\site-packages\rest_framework\decorators.py", line 50, in handler
    return func(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^
  File "C:\Other Project\Ads\Bitcoin Scraping\scrapperE\scrap\views.py", line 122, in save_bitstamp_minutely
    call_bitsamp_api(ohlc_url,minute_url,error_log_url)
  File "C:\Other Project\Ads\Bitcoin Scraping\scrapperE\scrap\views.py", line 80, in call_bitsamp_api
    "error_content": data.to_dict() if hasattr(data, 'to_dict') else str(data)
                                               ^^^^
UnboundLocalError: cannot access local variable 'data' where it is not associated with a value
2024-02-26 18:35:16,181 - DEBUG - Starting new HTTPS connection (1): www.bitstamp.net:443
2024-02-26 18:35:17,161 - DEBUG - https://www.bitstamp.net:443 "GET /api/v2/ohlc/btcusd/?step=3600&limit=1 HTTP/1.1" 200 None
2024-02-26 18:35:22,846 - DEBUG - Starting new HTTP connection (1): 127.0.0.1:8000
2024-02-26 18:35:22,860 - DEBUG - http://127.0.0.1:8000 "POST /scrap/api/bitstampminute/add HTTP/1.1" 200 327
2024-02-26 18:35:22,927 - INFO - Data successfully added to the database
2024-02-26 18:40:08,203 - DEBUG - Starting new HTTPS connection (1): www.bitstamp.net:443
2024-02-26 18:40:09,513 - DEBUG - https://www.bitstamp.net:443 "GET /api/v2/ohlc/btcusd/?step=3600&limit=1 HTTP/1.1" 200 None
2024-02-26 18:40:09,515 - DEBUG - Starting new HTTP connection (1): 127.0.0.1:8000
2024-02-26 18:40:09,542 - DEBUG - http://127.0.0.1:8000 "POST /scrap/api/bitstampminute/add HTTP/1.1" 200 327
2024-02-26 18:40:09,579 - INFO - Data successfully added to the database
2024-02-26 18:42:04,212 - INFO - C:\Other Project\Ads\Bitcoin Scraping\scrapperE\scrap\views.py changed, reloading.
2024-02-26 18:42:36,752 - DEBUG - Starting new HTTPS connection (1): www.bitstamp.net:443
2024-02-26 18:42:38,012 - DEBUG - https://www.bitstamp.net:443 "GET /api/v2/ohlc/btcusd/?step=3600&limit=1 HTTP/1.1" 200 None
2024-02-26 18:42:38,238 - DEBUG - Starting new HTTP connection (1): 127.0.0.1:8000
2024-02-26 18:42:38,250 - DEBUG - http://127.0.0.1:8000 "POST /scrap/api/bitstamp/add HTTP/1.1" 200 329
2024-02-26 18:42:38,250 - INFO - api_endpoint: http://127.0.0.1:8000/scrap/api/bitstamp/add status: 200 response: {"id":6784,"unix":1708941600,"date":"2024-02-26T10:00:00Z","local_date":"2024-02-26T18:00:00Z","symbol":"SCHEDULER_TEST","open":"50963.0000000000","high":"51233.0000000000","low":"50963.0000000000","close":"51208.0000000000","volume_btc":"37.6984905000","volume_usd":"1930464.3015240000","added_at":"2024-02-26T10:42:38.246150Z"}
2024-02-26 18:42:38,251 - INFO - Data successfully added to the database
2024-02-26 18:45:05,516 - DEBUG - Starting new HTTPS connection (1): www.bitstamp.net:443
2024-02-26 18:45:06,773 - DEBUG - https://www.bitstamp.net:443 "GET /api/v2/ohlc/btcusd/?step=3600&limit=1 HTTP/1.1" 200 None
2024-02-26 18:45:06,777 - DEBUG - Starting new HTTP connection (1): 127.0.0.1:8000
2024-02-26 18:45:06,797 - DEBUG - http://127.0.0.1:8000 "POST /scrap/api/bitstampminute/add HTTP/1.1" 200 327
2024-02-26 18:45:06,799 - INFO - api_endpoint: http://127.0.0.1:8000/scrap/api/bitstampminute/add status: 200 response: {"id":24,"unix":1708941600,"date":"2024-02-26T10:00:00Z","local_date":"2024-02-26T18:00:00Z","symbol":"SCHEDULER_TEST","open":"50963.0000000000","high":"51233.0000000000","low":"50963.0000000000","close":"51188.0000000000","volume_btc":"37.9130609200","volume_usd":"1940693.7623729599","added_at":"2024-02-26T10:45:06.793248Z"}
2024-02-26 18:45:06,799 - INFO - Data successfully added to the database
2024-02-26 18:50:05,302 - DEBUG - Starting new HTTPS connection (1): www.bitstamp.net:443
2024-02-26 18:50:07,043 - DEBUG - https://www.bitstamp.net:443 "GET /api/v2/ohlc/btcusd/?step=3600&limit=1 HTTP/1.1" 200 None
2024-02-26 18:50:07,048 - DEBUG - Starting new HTTP connection (1): 127.0.0.1:8000
2024-02-26 18:50:07,067 - DEBUG - http://127.0.0.1:8000 "POST /scrap/api/bitstampminute/add HTTP/1.1" 200 327
2024-02-26 18:50:07,068 - INFO - api_endpoint: http://127.0.0.1:8000/scrap/api/bitstampminute/add status: 200 response: {"id":25,"unix":1708941600,"date":"2024-02-26T10:00:00Z","local_date":"2024-02-26T18:00:00Z","symbol":"SCHEDULER_TEST","open":"50963.0000000000","high":"51233.0000000000","low":"50963.0000000000","close":"51229.0000000000","volume_btc":"38.4815785600","volume_usd":"1971372.7880502401","added_at":"2024-02-26T10:50:07.064675Z"}
2024-02-26 18:50:07,068 - INFO - Data successfully added to the database
2024-02-26 18:55:04,530 - DEBUG - Starting new HTTPS connection (1): www.bitstamp.net:443
2024-02-26 18:55:05,677 - DEBUG - https://www.bitstamp.net:443 "GET /api/v2/ohlc/btcusd/?step=3600&limit=1 HTTP/1.1" 200 None
2024-02-26 18:55:05,682 - DEBUG - Starting new HTTP connection (1): 127.0.0.1:8000
2024-02-26 18:55:05,704 - DEBUG - http://127.0.0.1:8000 "POST /scrap/api/bitstampminute/add HTTP/1.1" 200 327
2024-02-26 18:55:05,704 - INFO - api_endpoint: http://127.0.0.1:8000/scrap/api/bitstampminute/add status: 200 response: {"id":26,"unix":1708941600,"date":"2024-02-26T10:00:00Z","local_date":"2024-02-26T18:00:00Z","symbol":"SCHEDULER_TEST","open":"50963.0000000000","high":"51233.0000000000","low":"50963.0000000000","close":"51220.0000000000","volume_btc":"38.7626133200","volume_usd":"1985421.0542504000","added_at":"2024-02-26T10:55:05.699751Z"}
2024-02-26 18:55:05,705 - INFO - Data successfully added to the database
2024-02-27 06:29:18,693 - DEBUG - Starting new HTTPS connection (1): www.bitstamp.net:443
2024-02-27 06:29:18,870 - DEBUG - Starting new HTTPS connection (1): www.bitstamp.net:443
2024-02-27 06:29:20,134 - DEBUG - https://www.bitstamp.net:443 "GET /api/v2/ohlc/btcusd/?step=3600&limit=1 HTTP/1.1" 200 None
2024-02-27 06:29:20,136 - DEBUG - Starting new HTTP connection (1): 127.0.0.1:8000
2024-02-27 06:29:20,146 - DEBUG - http://127.0.0.1:8000 "POST /scrap/api/bitstamphour/add HTTP/1.1" 200 330
2024-02-27 06:29:20,147 - INFO - api_endpoint: http://127.0.0.1:8000/scrap/api/bitstamphour/add status: 200 response: {"id":86362,"unix":1708984800,"date":"2024-02-26T22:00:00Z","local_date":"2024-02-27T06:00:00Z","symbol":"SCHEDULER_TEST","open":"54679.0000000000","high":"54835.0000000000","low":"54543.0000000000","close":"54641.0000000000","volume_btc":"20.2012840100","volume_usd":"1103818.3595904100","added_at":"2024-02-26T22:29:20.143598Z"}
2024-02-27 06:29:20,147 - INFO - Data successfully added to the database
2024-02-27 06:29:20,242 - DEBUG - https://www.bitstamp.net:443 "GET /api/v2/ohlc/btcusd/?step=3600&limit=1 HTTP/1.1" 200 None
2024-02-27 06:29:20,244 - DEBUG - Starting new HTTP connection (1): 127.0.0.1:8000
2024-02-27 06:29:20,273 - DEBUG - http://127.0.0.1:8000 "POST /scrap/api/bitstampminute/add HTTP/1.1" 200 327
2024-02-27 06:29:20,275 - INFO - api_endpoint: http://127.0.0.1:8000/scrap/api/bitstampminute/add status: 200 response: {"id":27,"unix":1708984800,"date":"2024-02-26T22:00:00Z","local_date":"2024-02-27T06:00:00Z","symbol":"SCHEDULER_TEST","open":"54679.0000000000","high":"54835.0000000000","low":"54543.0000000000","close":"54641.0000000000","volume_btc":"20.2012840100","volume_usd":"1103818.3595904100","added_at":"2024-02-26T22:29:20.269989Z"}
2024-02-27 06:29:20,277 - INFO - Data successfully added to the database
2024-02-27 06:30:14,781 - DEBUG - Starting new HTTPS connection (1): www.bitstamp.net:443
2024-02-27 06:30:15,427 - DEBUG - https://www.bitstamp.net:443 "GET /api/v2/ohlc/btcusd/?step=3600&limit=1 HTTP/1.1" 200 None
2024-02-27 06:30:15,428 - DEBUG - Starting new HTTP connection (1): 127.0.0.1:8000
2024-02-27 06:30:15,441 - DEBUG - http://127.0.0.1:8000 "POST /scrap/api/bitstampminute/add HTTP/1.1" 200 327
2024-02-27 06:30:15,444 - INFO - api_endpoint: http://127.0.0.1:8000/scrap/api/bitstampminute/add status: 200 response: {"id":28,"unix":1708984800,"date":"2024-02-26T22:00:00Z","local_date":"2024-02-27T06:00:00Z","symbol":"SCHEDULER_TEST","open":"54679.0000000000","high":"54835.0000000000","low":"54543.0000000000","close":"54606.0000000000","volume_btc":"20.3715127300","volume_usd":"1112406.8241343799","added_at":"2024-02-26T22:30:15.438321Z"}
2024-02-27 06:30:15,449 - INFO - Data successfully added to the database
2024-02-27 07:17:38,959 - DEBUG - Starting new HTTPS connection (1): www.bitstamp.net:443
2024-02-27 07:17:39,816 - DEBUG - Starting new HTTPS connection (1): www.bitstamp.net:443
2024-02-27 07:17:40,038 - DEBUG - https://www.bitstamp.net:443 "GET /api/v2/ohlc/btcusd/?step=3600&limit=1 HTTP/1.1" 200 None
2024-02-27 07:17:40,040 - DEBUG - Starting new HTTP connection (1): 127.0.0.1:8000
2024-02-27 07:17:40,079 - DEBUG - http://127.0.0.1:8000 "POST /scrap/api/bitstamphour/add HTTP/1.1" 200 329
2024-02-27 07:17:40,086 - INFO - api_endpoint: http://127.0.0.1:8000/scrap/api/bitstamphour/add status: 200 response: {"id":86363,"unix":1708988400,"date":"2024-02-26T23:00:00Z","local_date":"2024-02-27T07:00:00Z","symbol":"SCHEDULER_TEST","open":"54710.0000000000","high":"54711.0000000000","low":"54552.0000000000","close":"54591.0000000000","volume_btc":"16.5665903300","volume_usd":"904386.7327050300","added_at":"2024-02-26T23:17:40.075796Z"}
2024-02-27 07:17:40,087 - INFO - Data successfully added to the database
2024-02-27 07:17:40,921 - DEBUG - https://www.bitstamp.net:443 "GET /api/v2/ohlc/btcusd/?step=3600&limit=1 HTTP/1.1" 200 None
2024-02-27 07:17:40,922 - DEBUG - Starting new HTTP connection (1): 127.0.0.1:8000
2024-02-27 07:17:40,965 - DEBUG - http://127.0.0.1:8000 "POST /scrap/api/bitstampminute/add HTTP/1.1" 200 326
2024-02-27 07:17:40,975 - INFO - api_endpoint: http://127.0.0.1:8000/scrap/api/bitstampminute/add status: 200 response: {"id":29,"unix":1708988400,"date":"2024-02-26T23:00:00Z","local_date":"2024-02-27T07:00:00Z","symbol":"SCHEDULER_TEST","open":"54710.0000000000","high":"54711.0000000000","low":"54552.0000000000","close":"54591.0000000000","volume_btc":"16.5665903300","volume_usd":"904386.7327050300","added_at":"2024-02-26T23:17:40.960569Z"}
2024-02-27 07:17:40,981 - INFO - Data successfully added to the database
2024-02-27 08:29:34,119 - DEBUG - Starting new HTTPS connection (1): www.bitstamp.net:443
2024-02-27 08:29:35,162 - DEBUG - https://www.bitstamp.net:443 "GET /api/v2/ohlc/btcusd/?step=3600&limit=1 HTTP/1.1" 200 None
2024-02-27 08:29:35,164 - DEBUG - Starting new HTTP connection (1): 127.0.0.1:8000
2024-02-27 08:29:35,276 - DEBUG - http://127.0.0.1:8000 "POST /scrap/api/bitstamphour/add HTTP/1.1" 200 330
2024-02-27 08:29:35,347 - INFO - api_endpoint: http://127.0.0.1:8000/scrap/api/bitstamphour/add status: 200 response: {"id":86364,"unix":1708992000,"date":"2024-02-27T00:00:00Z","local_date":"2024-02-27T08:00:00Z","symbol":"SCHEDULER_TEST","open":"54534.0000000000","high":"54715.0000000000","low":"54519.0000000000","close":"54658.0000000000","volume_btc":"27.3733571500","volume_usd":"1496172.9551047000","added_at":"2024-02-27T00:29:35.219286Z"}
2024-02-27 08:29:35,461 - INFO - Data successfully added to the database
2024-02-27 08:29:48,511 - DEBUG - Starting new HTTPS connection (1): www.bitstamp.net:443
2024-02-27 08:29:49,458 - DEBUG - https://www.bitstamp.net:443 "GET /api/v2/ohlc/btcusd/?step=3600&limit=1 HTTP/1.1" 200 None
2024-02-27 08:29:49,459 - DEBUG - Starting new HTTP connection (1): 127.0.0.1:8000
2024-02-27 08:29:49,467 - DEBUG - http://127.0.0.1:8000 "POST /scrap/api/bitstampminute/add HTTP/1.1" 200 327
2024-02-27 08:29:49,467 - INFO - api_endpoint: http://127.0.0.1:8000/scrap/api/bitstampminute/add status: 200 response: {"id":30,"unix":1708992000,"date":"2024-02-27T00:00:00Z","local_date":"2024-02-27T08:00:00Z","symbol":"SCHEDULER_TEST","open":"54534.0000000000","high":"54715.0000000000","low":"54519.0000000000","close":"54657.0000000000","volume_btc":"27.3757827900","volume_usd":"1496278.1599530298","added_at":"2024-02-27T00:29:49.465569Z"}
2024-02-27 08:29:49,468 - INFO - Data successfully added to the database
2024-02-27 08:30:10,029 - DEBUG - Starting new HTTPS connection (1): www.bitstamp.net:443
2024-02-27 08:30:10,635 - DEBUG - https://www.bitstamp.net:443 "GET /api/v2/ohlc/btcusd/?step=3600&limit=1 HTTP/1.1" 200 None
2024-02-27 08:30:10,647 - DEBUG - Starting new HTTP connection (1): 127.0.0.1:8000
2024-02-27 08:30:10,656 - DEBUG - http://127.0.0.1:8000 "POST /scrap/api/bitstampminute/add HTTP/1.1" 200 327
2024-02-27 08:30:10,656 - INFO - api_endpoint: http://127.0.0.1:8000/scrap/api/bitstampminute/add status: 200 response: {"id":31,"unix":1708992000,"date":"2024-02-27T00:00:00Z","local_date":"2024-02-27T08:00:00Z","symbol":"SCHEDULER_TEST","open":"54534.0000000000","high":"54715.0000000000","low":"54519.0000000000","close":"54661.0000000000","volume_btc":"27.4246417700","volume_usd":"1499058.3437899700","added_at":"2024-02-27T00:30:10.653645Z"}
2024-02-27 08:30:10,657 - INFO - Data successfully added to the database
2024-02-27 08:35:05,824 - DEBUG - Starting new HTTPS connection (1): www.bitstamp.net:443
2024-02-27 08:35:07,072 - DEBUG - https://www.bitstamp.net:443 "GET /api/v2/ohlc/btcusd/?step=3600&limit=1 HTTP/1.1" 200 None
2024-02-27 08:35:07,074 - DEBUG - Starting new HTTP connection (1): 127.0.0.1:8000
2024-02-27 08:35:07,088 - DEBUG - http://127.0.0.1:8000 "POST /scrap/api/bitstampminute/add HTTP/1.1" 200 327
2024-02-27 08:35:07,089 - INFO - api_endpoint: http://127.0.0.1:8000/scrap/api/bitstampminute/add status: 200 response: {"id":32,"unix":1708992000,"date":"2024-02-27T00:00:00Z","local_date":"2024-02-27T08:00:00Z","symbol":"SCHEDULER_TEST","open":"54534.0000000000","high":"54715.0000000000","low":"54519.0000000000","close":"54550.0000000000","volume_btc":"28.5843971900","volume_usd":"1559278.8667145001","added_at":"2024-02-27T00:35:07.084220Z"}
2024-02-27 08:35:07,089 - INFO - Data successfully added to the database
2024-02-27 08:40:05,229 - DEBUG - Starting new HTTPS connection (1): www.bitstamp.net:443
2024-02-27 08:40:06,393 - DEBUG - https://www.bitstamp.net:443 "GET /api/v2/ohlc/btcusd/?step=3600&limit=1 HTTP/1.1" 200 None
2024-02-27 08:40:06,398 - DEBUG - Starting new HTTP connection (1): 127.0.0.1:8000
2024-02-27 08:40:06,415 - DEBUG - http://127.0.0.1:8000 "POST /scrap/api/bitstampminute/add HTTP/1.1" 200 327
2024-02-27 08:40:06,417 - INFO - api_endpoint: http://127.0.0.1:8000/scrap/api/bitstampminute/add status: 200 response: {"id":33,"unix":1708992000,"date":"2024-02-27T00:00:00Z","local_date":"2024-02-27T08:00:00Z","symbol":"SCHEDULER_TEST","open":"54534.0000000000","high":"54715.0000000000","low":"54501.0000000000","close":"54519.0000000000","volume_btc":"32.3598763400","volume_usd":"1764228.0981804600","added_at":"2024-02-27T00:40:06.411797Z"}
2024-02-27 08:40:06,417 - INFO - Data successfully added to the database
2024-02-27 08:45:05,630 - DEBUG - Starting new HTTPS connection (1): www.bitstamp.net:443
2024-02-27 08:45:06,717 - DEBUG - https://www.bitstamp.net:443 "GET /api/v2/ohlc/btcusd/?step=3600&limit=1 HTTP/1.1" 200 None
2024-02-27 08:45:06,718 - DEBUG - Starting new HTTP connection (1): 127.0.0.1:8000
2024-02-27 08:45:06,730 - DEBUG - http://127.0.0.1:8000 "POST /scrap/api/bitstampminute/add HTTP/1.1" 200 327
2024-02-27 08:45:06,730 - INFO - api_endpoint: http://127.0.0.1:8000/scrap/api/bitstampminute/add status: 200 response: {"id":34,"unix":1708992000,"date":"2024-02-27T00:00:00Z","local_date":"2024-02-27T08:00:00Z","symbol":"SCHEDULER_TEST","open":"54534.0000000000","high":"54715.0000000000","low":"54500.0000000000","close":"54517.0000000000","volume_btc":"33.9000478000","volume_usd":"1848128.9059126002","added_at":"2024-02-27T00:45:06.726176Z"}
2024-02-27 08:45:06,730 - INFO - Data successfully added to the database
2024-02-27 08:50:32,541 - DEBUG - Starting new HTTPS connection (1): www.bitstamp.net:443
2024-02-27 08:50:33,720 - DEBUG - https://www.bitstamp.net:443 "GET /api/v2/ohlc/btcusd/?step=3600&limit=1 HTTP/1.1" 200 None
2024-02-27 08:50:33,722 - DEBUG - Starting new HTTP connection (1): 127.0.0.1:8000
2024-02-27 08:50:33,735 - DEBUG - http://127.0.0.1:8000 "POST /scrap/api/bitstampminute/add HTTP/1.1" 200 327
2024-02-27 08:50:33,736 - INFO - api_endpoint: http://127.0.0.1:8000/scrap/api/bitstampminute/add status: 200 response: {"id":35,"unix":1708992000,"date":"2024-02-27T00:00:00Z","local_date":"2024-02-27T08:00:00Z","symbol":"SCHEDULER_TEST","open":"54534.0000000000","high":"54715.0000000000","low":"54500.0000000000","close":"54577.0000000000","volume_btc":"35.0401707100","volume_usd":"1912387.3968396700","added_at":"2024-02-27T00:50:33.731003Z"}
2024-02-27 08:50:33,736 - INFO - Data successfully added to the database
2024-02-27 08:55:04,970 - DEBUG - Starting new HTTPS connection (1): www.bitstamp.net:443
2024-02-27 08:55:06,012 - DEBUG - https://www.bitstamp.net:443 "GET /api/v2/ohlc/btcusd/?step=3600&limit=1 HTTP/1.1" 200 None
2024-02-27 08:55:06,016 - DEBUG - Starting new HTTP connection (1): 127.0.0.1:8000
2024-02-27 08:55:06,037 - DEBUG - http://127.0.0.1:8000 "POST /scrap/api/bitstampminute/add HTTP/1.1" 200 327
2024-02-27 08:55:06,038 - INFO - api_endpoint: http://127.0.0.1:8000/scrap/api/bitstampminute/add status: 200 response: {"id":36,"unix":1708992000,"date":"2024-02-27T00:00:00Z","local_date":"2024-02-27T08:00:00Z","symbol":"SCHEDULER_TEST","open":"54534.0000000000","high":"54715.0000000000","low":"54500.0000000000","close":"54528.0000000000","volume_btc":"35.6972850400","volume_usd":"1946501.5586611198","added_at":"2024-02-27T00:55:06.030294Z"}
2024-02-27 08:55:06,039 - INFO - Data successfully added to the database
2024-02-27 09:00:05,264 - DEBUG - Starting new HTTPS connection (1): www.bitstamp.net:443
2024-02-27 09:00:06,172 - DEBUG - Starting new HTTPS connection (1): www.bitstamp.net:443
2024-02-27 09:00:06,732 - DEBUG - https://www.bitstamp.net:443 "GET /api/v2/ohlc/btcusd/?step=3600&limit=1 HTTP/1.1" 200 None
2024-02-27 09:00:06,735 - DEBUG - Starting new HTTP connection (1): 127.0.0.1:8000
2024-02-27 09:00:06,750 - DEBUG - http://127.0.0.1:8000 "POST /scrap/api/bitstampminute/add HTTP/1.1" 200 323
2024-02-27 09:00:06,751 - INFO - api_endpoint: http://127.0.0.1:8000/scrap/api/bitstampminute/add status: 200 response: {"id":37,"unix":1708995600,"date":"2024-02-27T01:00:00Z","local_date":"2024-02-27T09:00:00Z","symbol":"SCHEDULER_TEST","open":"54603.0000000000","high":"54604.0000000000","low":"54603.0000000000","close":"54604.0000000000","volume_btc":"0.1498889900","volume_usd":"8184.5384099600","added_at":"2024-02-27T01:00:06.745534Z"}
2024-02-27 09:00:06,751 - INFO - Data successfully added to the database
2024-02-27 09:00:07,313 - DEBUG - https://www.bitstamp.net:443 "GET /api/v2/ohlc/btcusd/?step=3600&limit=1 HTTP/1.1" 200 None
2024-02-27 09:00:07,318 - DEBUG - Starting new HTTP connection (1): 127.0.0.1:8000
2024-02-27 09:00:07,339 - DEBUG - http://127.0.0.1:8000 "POST /scrap/api/bitstamphour/add HTTP/1.1" 200 326
2024-02-27 09:00:07,341 - INFO - api_endpoint: http://127.0.0.1:8000/scrap/api/bitstamphour/add status: 200 response: {"id":86365,"unix":1708995600,"date":"2024-02-27T01:00:00Z","local_date":"2024-02-27T09:00:00Z","symbol":"SCHEDULER_TEST","open":"54603.0000000000","high":"54604.0000000000","low":"54603.0000000000","close":"54604.0000000000","volume_btc":"0.1498889900","volume_usd":"8184.5384099600","added_at":"2024-02-27T01:00:07.336036Z"}
2024-02-27 09:00:07,342 - INFO - Data successfully added to the database
2024-02-27 09:05:04,963 - DEBUG - Starting new HTTPS connection (1): www.bitstamp.net:443
2024-02-27 09:05:06,343 - DEBUG - https://www.bitstamp.net:443 "GET /api/v2/ohlc/btcusd/?step=3600&limit=1 HTTP/1.1" 200 None
2024-02-27 09:05:06,347 - DEBUG - Starting new HTTP connection (1): 127.0.0.1:8000
2024-02-27 09:05:06,369 - DEBUG - http://127.0.0.1:8000 "POST /scrap/api/bitstampminute/add HTTP/1.1" 200 325
2024-02-27 09:05:06,370 - INFO - api_endpoint: http://127.0.0.1:8000/scrap/api/bitstampminute/add status: 200 response: {"id":38,"unix":1708995600,"date":"2024-02-27T01:00:00Z","local_date":"2024-02-27T09:00:00Z","symbol":"SCHEDULER_TEST","open":"54603.0000000000","high":"54628.0000000000","low":"54564.0000000000","close":"54620.0000000000","volume_btc":"4.1413819300","volume_usd":"226202.2810166000","added_at":"2024-02-27T01:05:06.362555Z"}
2024-02-27 09:05:06,371 - INFO - Data successfully added to the database
2024-02-27 09:10:07,387 - DEBUG - Starting new HTTPS connection (1): www.bitstamp.net:443
2024-02-27 09:10:08,308 - DEBUG - https://www.bitstamp.net:443 "GET /api/v2/ohlc/btcusd/?step=3600&limit=1 HTTP/1.1" 200 None
2024-02-27 09:10:08,310 - DEBUG - Starting new HTTP connection (1): 127.0.0.1:8000
2024-02-27 09:10:08,322 - DEBUG - http://127.0.0.1:8000 "POST /scrap/api/bitstampminute/add HTTP/1.1" 200 327
2024-02-27 09:10:08,323 - INFO - api_endpoint: http://127.0.0.1:8000/scrap/api/bitstampminute/add status: 200 response: {"id":39,"unix":1708995600,"date":"2024-02-27T01:00:00Z","local_date":"2024-02-27T09:00:00Z","symbol":"SCHEDULER_TEST","open":"54603.0000000000","high":"54735.0000000000","low":"54564.0000000000","close":"54735.0000000000","volume_btc":"41.4192824500","volume_usd":"2267084.4249007497","added_at":"2024-02-27T01:10:08.318056Z"}
2024-02-27 09:10:08,323 - INFO - Data successfully added to the database
2024-02-27 09:15:05,068 - DEBUG - Starting new HTTPS connection (1): www.bitstamp.net:443
2024-02-27 09:15:06,053 - DEBUG - https://www.bitstamp.net:443 "GET /api/v2/ohlc/btcusd/?step=3600&limit=1 HTTP/1.1" 200 None
2024-02-27 09:15:06,055 - DEBUG - Starting new HTTP connection (1): 127.0.0.1:8000
2024-02-27 09:15:06,067 - DEBUG - http://127.0.0.1:8000 "POST /scrap/api/bitstampminute/add HTTP/1.1" 200 327
2024-02-27 09:15:06,067 - INFO - api_endpoint: http://127.0.0.1:8000/scrap/api/bitstampminute/add status: 200 response: {"id":40,"unix":1708995600,"date":"2024-02-27T01:00:00Z","local_date":"2024-02-27T09:00:00Z","symbol":"SCHEDULER_TEST","open":"54603.0000000000","high":"54758.0000000000","low":"54564.0000000000","close":"54725.0000000000","volume_btc":"45.9436038500","volume_usd":"2514263.7206912500","added_at":"2024-02-27T01:15:06.063430Z"}
2024-02-27 09:15:06,067 - INFO - Data successfully added to the database
2024-02-27 09:20:04,677 - DEBUG - Starting new HTTPS connection (1): www.bitstamp.net:443
2024-02-27 09:20:05,816 - DEBUG - https://www.bitstamp.net:443 "GET /api/v2/ohlc/btcusd/?step=3600&limit=1 HTTP/1.1" 200 None
2024-02-27 09:20:05,822 - DEBUG - Starting new HTTP connection (1): 127.0.0.1:8000
2024-02-27 09:20:05,840 - DEBUG - http://127.0.0.1:8000 "POST /scrap/api/bitstampminute/add HTTP/1.1" 200 327
2024-02-27 09:20:05,841 - INFO - api_endpoint: http://127.0.0.1:8000/scrap/api/bitstampminute/add status: 200 response: {"id":41,"unix":1708995600,"date":"2024-02-27T01:00:00Z","local_date":"2024-02-27T09:00:00Z","symbol":"SCHEDULER_TEST","open":"54603.0000000000","high":"54761.0000000000","low":"54564.0000000000","close":"54694.0000000000","volume_btc":"46.1292768800","volume_usd":"2522994.6696747200","added_at":"2024-02-27T01:20:05.834949Z"}
2024-02-27 09:20:05,841 - INFO - Data successfully added to the database
2024-02-27 09:25:04,647 - DEBUG - Starting new HTTPS connection (1): www.bitstamp.net:443
2024-02-27 09:25:05,754 - DEBUG - https://www.bitstamp.net:443 "GET /api/v2/ohlc/btcusd/?step=3600&limit=1 HTTP/1.1" 200 None
2024-02-27 09:25:05,756 - DEBUG - Starting new HTTP connection (1): 127.0.0.1:8000
2024-02-27 09:25:05,766 - DEBUG - http://127.0.0.1:8000 "POST /scrap/api/bitstampminute/add HTTP/1.1" 200 327
2024-02-27 09:25:05,766 - INFO - api_endpoint: http://127.0.0.1:8000/scrap/api/bitstampminute/add status: 200 response: {"id":42,"unix":1708995600,"date":"2024-02-27T01:00:00Z","local_date":"2024-02-27T09:00:00Z","symbol":"SCHEDULER_TEST","open":"54603.0000000000","high":"54761.0000000000","low":"54564.0000000000","close":"54741.0000000000","volume_btc":"47.3508240200","volume_usd":"2592031.4576788200","added_at":"2024-02-27T01:25:05.761864Z"}
2024-02-27 09:25:05,766 - INFO - Data successfully added to the database
2024-02-27 09:30:05,168 - DEBUG - Starting new HTTPS connection (1): www.bitstamp.net:443
2024-02-27 09:30:06,312 - DEBUG - https://www.bitstamp.net:443 "GET /api/v2/ohlc/btcusd/?step=3600&limit=1 HTTP/1.1" 200 None
2024-02-27 09:30:06,316 - DEBUG - Starting new HTTP connection (1): 127.0.0.1:8000
2024-02-27 09:30:06,335 - DEBUG - http://127.0.0.1:8000 "POST /scrap/api/bitstampminute/add HTTP/1.1" 200 327
2024-02-27 09:30:06,335 - INFO - api_endpoint: http://127.0.0.1:8000/scrap/api/bitstampminute/add status: 200 response: {"id":43,"unix":1708995600,"date":"2024-02-27T01:00:00Z","local_date":"2024-02-27T09:00:00Z","symbol":"SCHEDULER_TEST","open":"54603.0000000000","high":"54800.0000000000","low":"54564.0000000000","close":"54787.0000000000","volume_btc":"49.5158529100","volume_usd":"2712825.0333801700","added_at":"2024-02-27T01:30:06.329821Z"}
2024-02-27 09:30:06,336 - INFO - Data successfully added to the database
2024-02-27 09:35:04,763 - DEBUG - Starting new HTTPS connection (1): www.bitstamp.net:443
2024-02-27 09:35:05,700 - DEBUG - https://www.bitstamp.net:443 "GET /api/v2/ohlc/btcusd/?step=3600&limit=1 HTTP/1.1" 200 None
2024-02-27 09:35:05,703 - DEBUG - Starting new HTTP connection (1): 127.0.0.1:8000
2024-02-27 09:35:05,722 - DEBUG - http://127.0.0.1:8000 "POST /scrap/api/bitstampminute/add HTTP/1.1" 200 327
2024-02-27 09:35:05,726 - INFO - api_endpoint: http://127.0.0.1:8000/scrap/api/bitstampminute/add status: 200 response: {"id":44,"unix":1708995600,"date":"2024-02-27T01:00:00Z","local_date":"2024-02-27T09:00:00Z","symbol":"SCHEDULER_TEST","open":"54603.0000000000","high":"54802.0000000000","low":"54564.0000000000","close":"54776.0000000000","volume_btc":"50.1609491700","volume_usd":"2747616.1517359200","added_at":"2024-02-27T01:35:05.716793Z"}
2024-02-27 09:35:05,726 - INFO - Data successfully added to the database
2024-02-27 09:40:04,655 - DEBUG - Starting new HTTPS connection (1): www.bitstamp.net:443
2024-02-27 09:40:05,796 - DEBUG - https://www.bitstamp.net:443 "GET /api/v2/ohlc/btcusd/?step=3600&limit=1 HTTP/1.1" 200 None
2024-02-27 09:40:05,802 - DEBUG - Starting new HTTP connection (1): 127.0.0.1:8000
2024-02-27 09:40:05,823 - DEBUG - http://127.0.0.1:8000 "POST /scrap/api/bitstampminute/add HTTP/1.1" 200 327
2024-02-27 09:40:05,824 - INFO - api_endpoint: http://127.0.0.1:8000/scrap/api/bitstampminute/add status: 200 response: {"id":45,"unix":1708995600,"date":"2024-02-27T01:00:00Z","local_date":"2024-02-27T09:00:00Z","symbol":"SCHEDULER_TEST","open":"54603.0000000000","high":"54853.0000000000","low":"54564.0000000000","close":"54851.0000000000","volume_btc":"59.3999497700","volume_usd":"3258146.6448342700","added_at":"2024-02-27T01:40:05.816961Z"}
2024-02-27 09:40:05,824 - INFO - Data successfully added to the database
2024-02-27 09:45:05,005 - DEBUG - Starting new HTTPS connection (1): www.bitstamp.net:443
2024-02-27 09:45:06,190 - DEBUG - https://www.bitstamp.net:443 "GET /api/v2/ohlc/btcusd/?step=3600&limit=1 HTTP/1.1" 200 None
2024-02-27 09:45:06,192 - DEBUG - Starting new HTTP connection (1): 127.0.0.1:8000
2024-02-27 09:45:06,202 - DEBUG - http://127.0.0.1:8000 "POST /scrap/api/bitstampminute/add HTTP/1.1" 200 328
2024-02-27 09:45:06,202 - INFO - api_endpoint: http://127.0.0.1:8000/scrap/api/bitstampminute/add status: 200 response: {"id":46,"unix":1708995600,"date":"2024-02-27T01:00:00Z","local_date":"2024-02-27T09:00:00Z","symbol":"SCHEDULER_TEST","open":"54603.0000000000","high":"55307.0000000000","low":"54564.0000000000","close":"55141.0000000000","volume_btc":"123.5204082400","volume_usd":"6811038.8307618400","added_at":"2024-02-27T01:45:06.199354Z"}
2024-02-27 09:45:06,204 - INFO - Data successfully added to the database
2024-02-27 09:50:04,800 - DEBUG - Starting new HTTPS connection (1): www.bitstamp.net:443
2024-02-27 09:50:05,993 - DEBUG - https://www.bitstamp.net:443 "GET /api/v2/ohlc/btcusd/?step=3600&limit=1 HTTP/1.1" 200 None
2024-02-27 09:50:05,997 - DEBUG - Starting new HTTP connection (1): 127.0.0.1:8000
2024-02-27 09:50:06,017 - DEBUG - http://127.0.0.1:8000 "POST /scrap/api/bitstampminute/add HTTP/1.1" 200 329
2024-02-27 09:50:06,018 - INFO - api_endpoint: http://127.0.0.1:8000/scrap/api/bitstampminute/add status: 200 response: {"id":47,"unix":1708995600,"date":"2024-02-27T01:00:00Z","local_date":"2024-02-27T09:00:00Z","symbol":"SCHEDULER_TEST","open":"54603.0000000000","high":"55575.0000000000","low":"54564.0000000000","close":"55422.0000000000","volume_btc":"193.3718057500","volume_usd":"10717052.2182764990","added_at":"2024-02-27T01:50:06.011525Z"}
2024-02-27 09:50:06,018 - INFO - Data successfully added to the database
2024-02-27 09:55:04,617 - DEBUG - Starting new HTTPS connection (1): www.bitstamp.net:443
2024-02-27 09:55:05,858 - DEBUG - https://www.bitstamp.net:443 "GET /api/v2/ohlc/btcusd/?step=3600&limit=1 HTTP/1.1" 200 None
2024-02-27 09:55:05,860 - DEBUG - Starting new HTTP connection (1): 127.0.0.1:8000
2024-02-27 09:55:05,876 - DEBUG - http://127.0.0.1:8000 "POST /scrap/api/bitstampminute/add HTTP/1.1" 200 329
2024-02-27 09:55:05,877 - INFO - api_endpoint: http://127.0.0.1:8000/scrap/api/bitstampminute/add status: 200 response: {"id":48,"unix":1708995600,"date":"2024-02-27T01:00:00Z","local_date":"2024-02-27T09:00:00Z","symbol":"SCHEDULER_TEST","open":"54603.0000000000","high":"55723.5000000000","low":"54564.0000000000","close":"55660.0000000000","volume_btc":"210.5267271200","volume_usd":"11717917.6314992010","added_at":"2024-02-27T01:55:05.870926Z"}
2024-02-27 09:55:05,877 - INFO - Data successfully added to the database
2024-02-27 10:00:05,614 - DEBUG - Starting new HTTPS connection (1): www.bitstamp.net:443
2024-02-27 10:00:05,817 - DEBUG - Starting new HTTPS connection (1): www.bitstamp.net:443
2024-02-27 10:00:07,051 - DEBUG - https://www.bitstamp.net:443 "GET /api/v2/ohlc/btcusd/?step=3600&limit=1 HTTP/1.1" 200 None
2024-02-27 10:00:07,053 - DEBUG - Starting new HTTP connection (1): 127.0.0.1:8000
2024-02-27 10:00:07,063 - DEBUG - http://127.0.0.1:8000 "POST /scrap/api/bitstampminute/add HTTP/1.1" 200 323
2024-02-27 10:00:07,064 - INFO - api_endpoint: http://127.0.0.1:8000/scrap/api/bitstampminute/add status: 200 response: {"id":49,"unix":1708999200,"date":"2024-02-27T02:00:00Z","local_date":"2024-02-27T10:00:00Z","symbol":"SCHEDULER_TEST","open":"55814.0000000000","high":"55814.0000000000","low":"55814.0000000000","close":"55814.0000000000","volume_btc":"0.1000000000","volume_usd":"5581.4000000000","added_at":"2024-02-27T02:00:07.059931Z"}
2024-02-27 10:00:07,065 - INFO - Data successfully added to the database
2024-02-27 10:00:07,091 - DEBUG - Starting new HTTPS connection (1): www.bitstamp.net:443
2024-02-27 10:00:07,185 - DEBUG - https://www.bitstamp.net:443 "GET /api/v2/ohlc/btcusd/?step=3600&limit=1 HTTP/1.1" 200 None
2024-02-27 10:00:07,187 - DEBUG - Starting new HTTP connection (1): 127.0.0.1:8000
2024-02-27 10:00:07,259 - DEBUG - http://127.0.0.1:8000 "POST /scrap/api/bitstamp/add HTTP/1.1" 200 325
2024-02-27 10:00:07,260 - INFO - api_endpoint: http://127.0.0.1:8000/scrap/api/bitstamp/add status: 200 response: {"id":6785,"unix":1708999200,"date":"2024-02-27T02:00:00Z","local_date":"2024-02-27T10:00:00Z","symbol":"SCHEDULER_TEST","open":"55814.0000000000","high":"55814.0000000000","low":"55814.0000000000","close":"55814.0000000000","volume_btc":"0.1000000000","volume_usd":"5581.4000000000","added_at":"2024-02-27T02:00:07.256348Z"}
2024-02-27 10:00:07,260 - INFO - Data successfully added to the database
2024-02-27 10:00:07,696 - DEBUG - https://www.bitstamp.net:443 "GET /api/v2/ohlc/btcusd/?step=3600&limit=1 HTTP/1.1" 200 None
2024-02-27 10:00:07,700 - DEBUG - Starting new HTTP connection (1): 127.0.0.1:8000
2024-02-27 10:00:07,721 - DEBUG - http://127.0.0.1:8000 "POST /scrap/api/bitstamphour/add HTTP/1.1" 200 326
2024-02-27 10:00:07,727 - INFO - api_endpoint: http://127.0.0.1:8000/scrap/api/bitstamphour/add status: 200 response: {"id":86366,"unix":1708999200,"date":"2024-02-27T02:00:00Z","local_date":"2024-02-27T10:00:00Z","symbol":"SCHEDULER_TEST","open":"55814.0000000000","high":"55814.0000000000","low":"55814.0000000000","close":"55814.0000000000","volume_btc":"0.1000000000","volume_usd":"5581.4000000000","added_at":"2024-02-27T02:00:07.716874Z"}
2024-02-27 10:00:07,728 - INFO - Data successfully added to the database
2024-02-27 10:05:06,861 - DEBUG - Starting new HTTPS connection (1): www.bitstamp.net:443
2024-02-27 10:05:08,030 - DEBUG - https://www.bitstamp.net:443 "GET /api/v2/ohlc/btcusd/?step=3600&limit=1 HTTP/1.1" 200 None
2024-02-27 10:05:08,032 - DEBUG - Starting new HTTP connection (1): 127.0.0.1:8000
2024-02-27 10:05:08,044 - DEBUG - http://127.0.0.1:8000 "POST /scrap/api/bitstampminute/add HTTP/1.1" 200 327
2024-02-27 10:05:08,045 - INFO - api_endpoint: http://127.0.0.1:8000/scrap/api/bitstampminute/add status: 200 response: {"id":50,"unix":1708999200,"date":"2024-02-27T02:00:00Z","local_date":"2024-02-27T10:00:00Z","symbol":"SCHEDULER_TEST","open":"55814.0000000000","high":"55826.0000000000","low":"55645.0000000000","close":"55753.0000000000","volume_btc":"17.9813095000","volume_usd":"1002511.9485534999","added_at":"2024-02-27T02:05:08.039598Z"}
2024-02-27 10:05:08,045 - INFO - Data successfully added to the database
2024-02-27 10:11:24,049 - INFO - C:\Other Project\Ads\Bitcoin Scraping\scrapperE\scrap\views.py changed, reloading.
2024-02-27 10:11:40,736 - INFO - C:\Other Project\Ads\Bitcoin Scraping\scrapperE\scrap\views.py changed, reloading.
2024-02-27 10:13:03,919 - INFO - C:\Other Project\Ads\Bitcoin Scraping\scrapperE\scrap\views.py changed, reloading.
2024-02-27 10:14:32,145 - INFO - C:\Other Project\Ads\Bitcoin Scraping\scrapperE\scrapperE\urls.py changed, reloading.
2024-02-27 10:14:49,822 - INFO - C:\Other Project\Ads\Bitcoin Scraping\scrapperE\scrap\urls.py changed, reloading.
2024-02-27 10:28:31,150 - INFO - C:\Other Project\Ads\Bitcoin Scraping\scrapperE\scrapperE\settings.py changed, reloading.
2024-02-27 10:46:02,716 - INFO - C:\Other Project\Ads\Bitcoin Scraping\scrapperE\scrapperE\urls.py changed, reloading.
2024-02-27 11:18:14,757 - WARNING - Not Found: /scrap/pivot_data/
2024-02-27 11:18:22,976 - WARNING - Not Found: /scrap/data/2024-02-27 13:50:18,546 - INFO - C:\Other Project\Ads\Bitcoin Scraping\scrapperE\scrap\views.py changed, reloading.
2024-02-27 13:51:05,652 - INFO - C:\Other Project\Ads\Bitcoin Scraping\scrapperE\scrap\urls.py changed, reloading.
2024-02-27 16:12:25,006 - INFO - C:\Other Project\Ads\Bitcoin Scraping\scrapperE\scrap\views.py changed, reloading.
2024-02-27 16:12:32,281 - INFO - C:\Other Project\Ads\Bitcoin Scraping\scrapperE\scrap\urls.py changed, reloading.
2024-02-27 16:12:55,321 - INFO - C:\Other Project\Ads\Bitcoin Scraping\scrapperE\scrap\urls.py changed, reloading.
2024-02-27 16:13:07,611 - INFO - C:\Other Project\Ads\Bitcoin Scraping\scrapperE\scrap\urls.py changed, reloading.
2024-02-27 16:13:25,159 - ERROR - Internal Server Error: /scrap/
Traceback (most recent call last):
  File "C:\Users\Zhong_ming_Tan\AppData\Local\anaconda3\envs\fws-dell\Lib\site-packages\django\core\handlers\exception.py", line 55, in inner
    response = get_response(request)
               ^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Zhong_ming_Tan\AppData\Local\anaconda3\envs\fws-dell\Lib\site-packages\django\core\handlers\base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Other Project\Ads\Bitcoin Scraping\scrapperE\scrap\views.py", line 31, in dashboard_with_pivot
    return render(request, 'dashboard_with_pivot.html', {})
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Zhong_ming_Tan\AppData\Local\anaconda3\envs\fws-dell\Lib\site-packages\django\shortcuts.py", line 25, in render
    content = loader.render_to_string(template_name, context, request, using=using)
              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Zhong_ming_Tan\AppData\Local\anaconda3\envs\fws-dell\Lib\site-packages\django\template\loader.py", line 62, in render_to_string
    return template.render(context, request)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Zhong_ming_Tan\AppData\Local\anaconda3\envs\fws-dell\Lib\site-packages\django\template\backends\django.py", line 61, in render
    return self.template.render(context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Zhong_ming_Tan\AppData\Local\anaconda3\envs\fws-dell\Lib\site-packages\django\template\base.py", line 171, in render
    return self._render(context)
           ^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Zhong_ming_Tan\AppData\Local\anaconda3\envs\fws-dell\Lib\site-packages\django\template\base.py", line 163, in _render
    return self.nodelist.render(context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Zhong_ming_Tan\AppData\Local\anaconda3\envs\fws-dell\Lib\site-packages\django\template\base.py", line 1000, in render
    return SafeString("".join([node.render_annotated(context) for node in self]))
                               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Zhong_ming_Tan\AppData\Local\anaconda3\envs\fws-dell\Lib\site-packages\django\template\base.py", line 961, in render_annotated
    return self.render(context)
           ^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Zhong_ming_Tan\AppData\Local\anaconda3\envs\fws-dell\Lib\site-packages\django\template\defaulttags.py", line 479, in render
    url = reverse(view_name, args=args, kwargs=kwargs, current_app=current_app)
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Zhong_ming_Tan\AppData\Local\anaconda3\envs\fws-dell\Lib\site-packages\django\urls\base.py", line 88, in reverse
    return resolver._reverse_with_prefix(view, prefix, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Zhong_ming_Tan\AppData\Local\anaconda3\envs\fws-dell\Lib\site-packages\django\urls\resolvers.py", line 851, in _reverse_with_prefix
    raise NoReverseMatch(msg)
django.urls.exceptions.NoReverseMatch: Reverse for 'bitcoin_data_daily/get_all/' not found. 'bitcoin_data_daily/get_all/' is not a valid view function or pattern name.
2024-02-27 16:14:14,216 - ERROR - Internal Server Error: /scrap/
Traceback (most recent call last):
  File "C:\Users\Zhong_ming_Tan\AppData\Local\anaconda3\envs\fws-dell\Lib\site-packages\django\core\handlers\exception.py", line 55, in inner
    response = get_response(request)
               ^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Zhong_ming_Tan\AppData\Local\anaconda3\envs\fws-dell\Lib\site-packages\django\core\handlers\base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Other Project\Ads\Bitcoin Scraping\scrapperE\scrap\views.py", line 31, in dashboard_with_pivot
    return render(request, 'dashboard_with_pivot.html', {})
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Zhong_ming_Tan\AppData\Local\anaconda3\envs\fws-dell\Lib\site-packages\django\shortcuts.py", line 25, in render
    content = loader.render_to_string(template_name, context, request, using=using)
              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Zhong_ming_Tan\AppData\Local\anaconda3\envs\fws-dell\Lib\site-packages\django\template\loader.py", line 62, in render_to_string
    return template.render(context, request)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Zhong_ming_Tan\AppData\Local\anaconda3\envs\fws-dell\Lib\site-packages\django\template\backends\django.py", line 61, in render
    return self.template.render(context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Zhong_ming_Tan\AppData\Local\anaconda3\envs\fws-dell\Lib\site-packages\django\template\base.py", line 171, in render
    return self._render(context)
           ^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Zhong_ming_Tan\AppData\Local\anaconda3\envs\fws-dell\Lib\site-packages\django\template\base.py", line 163, in _render
    return self.nodelist.render(context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Zhong_ming_Tan\AppData\Local\anaconda3\envs\fws-dell\Lib\site-packages\django\template\base.py", line 1000, in render
    return SafeString("".join([node.render_annotated(context) for node in self]))
                               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Zhong_ming_Tan\AppData\Local\anaconda3\envs\fws-dell\Lib\site-packages\django\template\base.py", line 961, in render_annotated
    return self.render(context)
           ^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Zhong_ming_Tan\AppData\Local\anaconda3\envs\fws-dell\Lib\site-packages\django\template\defaulttags.py", line 479, in render
    url = reverse(view_name, args=args, kwargs=kwargs, current_app=current_app)
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Zhong_ming_Tan\AppData\Local\anaconda3\envs\fws-dell\Lib\site-packages\django\urls\base.py", line 88, in reverse
    return resolver._reverse_with_prefix(view, prefix, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Zhong_ming_Tan\AppData\Local\anaconda3\envs\fws-dell\Lib\site-packages\django\urls\resolvers.py", line 851, in _reverse_with_prefix
    raise NoReverseMatch(msg)
django.urls.exceptions.NoReverseMatch: Reverse for 'bitcoin_data_daily/get_all/' not found. 'bitcoin_data_daily/get_all/' is not a valid view function or pattern name.
2024-02-27 16:14:17,823 - ERROR - Internal Server Error: /scrap/
Traceback (most recent call last):
  File "C:\Users\Zhong_ming_Tan\AppData\Local\anaconda3\envs\fws-dell\Lib\site-packages\django\core\handlers\exception.py", line 55, in inner
    response = get_response(request)
               ^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Zhong_ming_Tan\AppData\Local\anaconda3\envs\fws-dell\Lib\site-packages\django\core\handlers\base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Other Project\Ads\Bitcoin Scraping\scrapperE\scrap\views.py", line 31, in dashboard_with_pivot
    return render(request, 'dashboard_with_pivot.html', {})
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Zhong_ming_Tan\AppData\Local\anaconda3\envs\fws-dell\Lib\site-packages\django\shortcuts.py", line 25, in render
    content = loader.render_to_string(template_name, context, request, using=using)
              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Zhong_ming_Tan\AppData\Local\anaconda3\envs\fws-dell\Lib\site-packages\django\template\loader.py", line 62, in render_to_string
    return template.render(context, request)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Zhong_ming_Tan\AppData\Local\anaconda3\envs\fws-dell\Lib\site-packages\django\template\backends\django.py", line 61, in render
    return self.template.render(context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Zhong_ming_Tan\AppData\Local\anaconda3\envs\fws-dell\Lib\site-packages\django\template\base.py", line 171, in render
    return self._render(context)
           ^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Zhong_ming_Tan\AppData\Local\anaconda3\envs\fws-dell\Lib\site-packages\django\template\base.py", line 163, in _render
    return self.nodelist.render(context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Zhong_ming_Tan\AppData\Local\anaconda3\envs\fws-dell\Lib\site-packages\django\template\base.py", line 1000, in render
    return SafeString("".join([node.render_annotated(context) for node in self]))
                               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Zhong_ming_Tan\AppData\Local\anaconda3\envs\fws-dell\Lib\site-packages\django\template\base.py", line 961, in render_annotated
    return self.render(context)
           ^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Zhong_ming_Tan\AppData\Local\anaconda3\envs\fws-dell\Lib\site-packages\django\template\defaulttags.py", line 479, in render
    url = reverse(view_name, args=args, kwargs=kwargs, current_app=current_app)
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Zhong_ming_Tan\AppData\Local\anaconda3\envs\fws-dell\Lib\site-packages\django\urls\base.py", line 88, in reverse
    return resolver._reverse_with_prefix(view, prefix, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Zhong_ming_Tan\AppData\Local\anaconda3\envs\fws-dell\Lib\site-packages\django\urls\resolvers.py", line 851, in _reverse_with_prefix
    raise NoReverseMatch(msg)
django.urls.exceptions.NoReverseMatch: Reverse for 'bitcoin_data_daily/get_all/' not found. 'bitcoin_data_daily/get_all/' is not a valid view function or pattern name.
2024-02-27 16:14:19,068 - ERROR - Internal Server Error: /scrap/
Traceback (most recent call last):
  File "C:\Users\Zhong_ming_Tan\AppData\Local\anaconda3\envs\fws-dell\Lib\site-packages\django\core\handlers\exception.py", line 55, in inner
    response = get_response(request)
               ^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Zhong_ming_Tan\AppData\Local\anaconda3\envs\fws-dell\Lib\site-packages\django\core\handlers\base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Other Project\Ads\Bitcoin Scraping\scrapperE\scrap\views.py", line 31, in dashboard_with_pivot
    return render(request, 'dashboard_with_pivot.html', {})
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Zhong_ming_Tan\AppData\Local\anaconda3\envs\fws-dell\Lib\site-packages\django\shortcuts.py", line 25, in render
    content = loader.render_to_string(template_name, context, request, using=using)
              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Zhong_ming_Tan\AppData\Local\anaconda3\envs\fws-dell\Lib\site-packages\django\template\loader.py", line 62, in render_to_string
    return template.render(context, request)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Zhong_ming_Tan\AppData\Local\anaconda3\envs\fws-dell\Lib\site-packages\django\template\backends\django.py", line 61, in render
    return self.template.render(context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Zhong_ming_Tan\AppData\Local\anaconda3\envs\fws-dell\Lib\site-packages\django\template\base.py", line 171, in render
    return self._render(context)
           ^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Zhong_ming_Tan\AppData\Local\anaconda3\envs\fws-dell\Lib\site-packages\django\template\base.py", line 163, in _render
    return self.nodelist.render(context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Zhong_ming_Tan\AppData\Local\anaconda3\envs\fws-dell\Lib\site-packages\django\template\base.py", line 1000, in render
    return SafeString("".join([node.render_annotated(context) for node in self]))
                               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Zhong_ming_Tan\AppData\Local\anaconda3\envs\fws-dell\Lib\site-packages\django\template\base.py", line 961, in render_annotated
    return self.render(context)
           ^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Zhong_ming_Tan\AppData\Local\anaconda3\envs\fws-dell\Lib\site-packages\django\template\defaulttags.py", line 479, in render
    url = reverse(view_name, args=args, kwargs=kwargs, current_app=current_app)
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Zhong_ming_Tan\AppData\Local\anaconda3\envs\fws-dell\Lib\site-packages\django\urls\base.py", line 88, in reverse
    return resolver._reverse_with_prefix(view, prefix, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Zhong_ming_Tan\AppData\Local\anaconda3\envs\fws-dell\Lib\site-packages\django\urls\resolvers.py", line 851, in _reverse_with_prefix
    raise NoReverseMatch(msg)
django.urls.exceptions.NoReverseMatch: Reverse for 'bitcoin_data_daily/get_all/' not found. 'bitcoin_data_daily/get_all/' is not a valid view function or pattern name.
2024-02-27 16:14:19,778 - ERROR - Internal Server Error: /scrap/
Traceback (most recent call last):
  File "C:\Users\Zhong_ming_Tan\AppData\Local\anaconda3\envs\fws-dell\Lib\site-packages\django\core\handlers\exception.py", line 55, in inner
    response = get_response(request)
               ^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Zhong_ming_Tan\AppData\Local\anaconda3\envs\fws-dell\Lib\site-packages\django\core\handlers\base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Other Project\Ads\Bitcoin Scraping\scrapperE\scrap\views.py", line 31, in dashboard_with_pivot
    return render(request, 'dashboard_with_pivot.html', {})
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Zhong_ming_Tan\AppData\Local\anaconda3\envs\fws-dell\Lib\site-packages\django\shortcuts.py", line 25, in render
    content = loader.render_to_string(template_name, context, request, using=using)
              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Zhong_ming_Tan\AppData\Local\anaconda3\envs\fws-dell\Lib\site-packages\django\template\loader.py", line 62, in render_to_string
    return template.render(context, request)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Zhong_ming_Tan\AppData\Local\anaconda3\envs\fws-dell\Lib\site-packages\django\template\backends\django.py", line 61, in render
    return self.template.render(context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Zhong_ming_Tan\AppData\Local\anaconda3\envs\fws-dell\Lib\site-packages\django\template\base.py", line 171, in render
    return self._render(context)
           ^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Zhong_ming_Tan\AppData\Local\anaconda3\envs\fws-dell\Lib\site-packages\django\template\base.py", line 163, in _render
    return self.nodelist.render(context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Zhong_ming_Tan\AppData\Local\anaconda3\envs\fws-dell\Lib\site-packages\django\template\base.py", line 1000, in render
    return SafeString("".join([node.render_annotated(context) for node in self]))
                               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Zhong_ming_Tan\AppData\Local\anaconda3\envs\fws-dell\Lib\site-packages\django\template\base.py", line 961, in render_annotated
    return self.render(context)
           ^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Zhong_ming_Tan\AppData\Local\anaconda3\envs\fws-dell\Lib\site-packages\django\template\defaulttags.py", line 479, in render
    url = reverse(view_name, args=args, kwargs=kwargs, current_app=current_app)
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Zhong_ming_Tan\AppData\Local\anaconda3\envs\fws-dell\Lib\site-packages\django\urls\base.py", line 88, in reverse
    return resolver._reverse_with_prefix(view, prefix, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Zhong_ming_Tan\AppData\Local\anaconda3\envs\fws-dell\Lib\site-packages\django\urls\resolvers.py", line 851, in _reverse_with_prefix
    raise NoReverseMatch(msg)
django.urls.exceptions.NoReverseMatch: Reverse for 'bitcoin_data_daily/get_all/' not found. 'bitcoin_data_daily/get_all/' is not a valid view function or pattern name.
2024-02-27 16:14:57,140 - INFO - C:\Other Project\Ads\Bitcoin Scraping\scrapperE\scrap\urls.py changed, reloading.
2024-02-27 16:15:03,826 - ERROR - Internal Server Error: /scrap/
Traceback (most recent call last):
  File "C:\Users\Zhong_ming_Tan\AppData\Local\anaconda3\envs\fws-dell\Lib\site-packages\django\core\handlers\exception.py", line 55, in inner
    response = get_response(request)
               ^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Zhong_ming_Tan\AppData\Local\anaconda3\envs\fws-dell\Lib\site-packages\django\core\handlers\base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Other Project\Ads\Bitcoin Scraping\scrapperE\scrap\views.py", line 31, in dashboard_with_pivot
    return render(request, 'dashboard_with_pivot.html', {})
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Zhong_ming_Tan\AppData\Local\anaconda3\envs\fws-dell\Lib\site-packages\django\shortcuts.py", line 25, in render
    content = loader.render_to_string(template_name, context, request, using=using)
              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Zhong_ming_Tan\AppData\Local\anaconda3\envs\fws-dell\Lib\site-packages\django\template\loader.py", line 62, in render_to_string
    return template.render(context, request)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Zhong_ming_Tan\AppData\Local\anaconda3\envs\fws-dell\Lib\site-packages\django\template\backends\django.py", line 61, in render
    return self.template.render(context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Zhong_ming_Tan\AppData\Local\anaconda3\envs\fws-dell\Lib\site-packages\django\template\base.py", line 171, in render
    return self._render(context)
           ^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Zhong_ming_Tan\AppData\Local\anaconda3\envs\fws-dell\Lib\site-packages\django\template\base.py", line 163, in _render
    return self.nodelist.render(context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Zhong_ming_Tan\AppData\Local\anaconda3\envs\fws-dell\Lib\site-packages\django\template\base.py", line 1000, in render
    return SafeString("".join([node.render_annotated(context) for node in self]))
                               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Zhong_ming_Tan\AppData\Local\anaconda3\envs\fws-dell\Lib\site-packages\django\template\base.py", line 961, in render_annotated
    return self.render(context)
           ^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Zhong_ming_Tan\AppData\Local\anaconda3\envs\fws-dell\Lib\site-packages\django\template\defaulttags.py", line 479, in render
    url = reverse(view_name, args=args, kwargs=kwargs, current_app=current_app)
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Zhong_ming_Tan\AppData\Local\anaconda3\envs\fws-dell\Lib\site-packages\django\urls\base.py", line 88, in reverse
    return resolver._reverse_with_prefix(view, prefix, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Zhong_ming_Tan\AppData\Local\anaconda3\envs\fws-dell\Lib\site-packages\django\urls\resolvers.py", line 851, in _reverse_with_prefix
    raise NoReverseMatch(msg)
django.urls.exceptions.NoReverseMatch: Reverse for 'bitcoin_data_daily/get_all' not found. 'bitcoin_data_daily/get_all' is not a valid view function or pattern name.
2024-02-27 16:15:28,641 - WARNING - Not Found: /scrap/bitcoin_data_daily/get_all/
2024-02-27 16:15:34,511 - ERROR - Internal Server Error: /scrap/
Traceback (most recent call last):
  File "C:\Users\Zhong_ming_Tan\AppData\Local\anaconda3\envs\fws-dell\Lib\site-packages\django\core\handlers\exception.py", line 55, in inner
    response = get_response(request)
               ^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Zhong_ming_Tan\AppData\Local\anaconda3\envs\fws-dell\Lib\site-packages\django\core\handlers\base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Other Project\Ads\Bitcoin Scraping\scrapperE\scrap\views.py", line 31, in dashboard_with_pivot
    return render(request, 'dashboard_with_pivot.html', {})
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Zhong_ming_Tan\AppData\Local\anaconda3\envs\fws-dell\Lib\site-packages\django\shortcuts.py", line 25, in render
    content = loader.render_to_string(template_name, context, request, using=using)
              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Zhong_ming_Tan\AppData\Local\anaconda3\envs\fws-dell\Lib\site-packages\django\template\loader.py", line 62, in render_to_string
    return template.render(context, request)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Zhong_ming_Tan\AppData\Local\anaconda3\envs\fws-dell\Lib\site-packages\django\template\backends\django.py", line 61, in render
    return self.template.render(context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Zhong_ming_Tan\AppData\Local\anaconda3\envs\fws-dell\Lib\site-packages\django\template\base.py", line 171, in render
    return self._render(context)
           ^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Zhong_ming_Tan\AppData\Local\anaconda3\envs\fws-dell\Lib\site-packages\django\template\base.py", line 163, in _render
    return self.nodelist.render(context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Zhong_ming_Tan\AppData\Local\anaconda3\envs\fws-dell\Lib\site-packages\django\template\base.py", line 1000, in render
    return SafeString("".join([node.render_annotated(context) for node in self]))
                               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Zhong_ming_Tan\AppData\Local\anaconda3\envs\fws-dell\Lib\site-packages\django\template\base.py", line 961, in render_annotated
    return self.render(context)
           ^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Zhong_ming_Tan\AppData\Local\anaconda3\envs\fws-dell\Lib\site-packages\django\template\defaulttags.py", line 479, in render
    url = reverse(view_name, args=args, kwargs=kwargs, current_app=current_app)
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Zhong_ming_Tan\AppData\Local\anaconda3\envs\fws-dell\Lib\site-packages\django\urls\base.py", line 88, in reverse
    return resolver._reverse_with_prefix(view, prefix, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Zhong_ming_Tan\AppData\Local\anaconda3\envs\fws-dell\Lib\site-packages\django\urls\resolvers.py", line 851, in _reverse_with_prefix
    raise NoReverseMatch(msg)
django.urls.exceptions.NoReverseMatch: Reverse for '/bitcoin_data_daily/get_all' not found. '/bitcoin_data_daily/get_all' is not a valid view function or pattern name.
2024-02-27 16:15:57,204 - ERROR - Internal Server Error: /scrap/
Traceback (most recent call last):
  File "C:\Users\Zhong_ming_Tan\AppData\Local\anaconda3\envs\fws-dell\Lib\site-packages\django\core\handlers\exception.py", line 55, in inner
    response = get_response(request)
               ^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Zhong_ming_Tan\AppData\Local\anaconda3\envs\fws-dell\Lib\site-packages\django\core\handlers\base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Other Project\Ads\Bitcoin Scraping\scrapperE\scrap\views.py", line 31, in dashboard_with_pivot
    return render(request, 'dashboard_with_pivot.html', {})
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Zhong_ming_Tan\AppData\Local\anaconda3\envs\fws-dell\Lib\site-packages\django\shortcuts.py", line 25, in render
    content = loader.render_to_string(template_name, context, request, using=using)
              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Zhong_ming_Tan\AppData\Local\anaconda3\envs\fws-dell\Lib\site-packages\django\template\loader.py", line 62, in render_to_string
    return template.render(context, request)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Zhong_ming_Tan\AppData\Local\anaconda3\envs\fws-dell\Lib\site-packages\django\template\backends\django.py", line 61, in render
    return self.template.render(context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Zhong_ming_Tan\AppData\Local\anaconda3\envs\fws-dell\Lib\site-packages\django\template\base.py", line 171, in render
    return self._render(context)
           ^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Zhong_ming_Tan\AppData\Local\anaconda3\envs\fws-dell\Lib\site-packages\django\template\base.py", line 163, in _render
    return self.nodelist.render(context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Zhong_ming_Tan\AppData\Local\anaconda3\envs\fws-dell\Lib\site-packages\django\template\base.py", line 1000, in render
    return SafeString("".join([node.render_annotated(context) for node in self]))
                               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Zhong_ming_Tan\AppData\Local\anaconda3\envs\fws-dell\Lib\site-packages\django\template\base.py", line 961, in render_annotated
    return self.render(context)
           ^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Zhong_ming_Tan\AppData\Local\anaconda3\envs\fws-dell\Lib\site-packages\django\template\defaulttags.py", line 479, in render
    url = reverse(view_name, args=args, kwargs=kwargs, current_app=current_app)
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Zhong_ming_Tan\AppData\Local\anaconda3\envs\fws-dell\Lib\site-packages\django\urls\base.py", line 88, in reverse
    return resolver._reverse_with_prefix(view, prefix, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Zhong_ming_Tan\AppData\Local\anaconda3\envs\fws-dell\Lib\site-packages\django\urls\resolvers.py", line 851, in _reverse_with_prefix
    raise NoReverseMatch(msg)
django.urls.exceptions.NoReverseMatch: Reverse for 'bitcoin_data_daily/get_all' not found. 'bitcoin_data_daily/get_all' is not a valid view function or pattern name.
2024-02-27 16:16:00,074 - ERROR - Internal Server Error: /scrap/
Traceback (most recent call last):
  File "C:\Users\Zhong_ming_Tan\AppData\Local\anaconda3\envs\fws-dell\Lib\site-packages\django\core\handlers\exception.py", line 55, in inner
    response = get_response(request)
               ^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Zhong_ming_Tan\AppData\Local\anaconda3\envs\fws-dell\Lib\site-packages\django\core\handlers\base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Other Project\Ads\Bitcoin Scraping\scrapperE\scrap\views.py", line 31, in dashboard_with_pivot
    return render(request, 'dashboard_with_pivot.html', {})
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Zhong_ming_Tan\AppData\Local\anaconda3\envs\fws-dell\Lib\site-packages\django\shortcuts.py", line 25, in render
    content = loader.render_to_string(template_name, context, request, using=using)
              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Zhong_ming_Tan\AppData\Local\anaconda3\envs\fws-dell\Lib\site-packages\django\template\loader.py", line 62, in render_to_string
    return template.render(context, request)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Zhong_ming_Tan\AppData\Local\anaconda3\envs\fws-dell\Lib\site-packages\django\template\backends\django.py", line 61, in render
    return self.template.render(context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Zhong_ming_Tan\AppData\Local\anaconda3\envs\fws-dell\Lib\site-packages\django\template\base.py", line 171, in render
    return self._render(context)
           ^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Zhong_ming_Tan\AppData\Local\anaconda3\envs\fws-dell\Lib\site-packages\django\template\base.py", line 163, in _render
    return self.nodelist.render(context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Zhong_ming_Tan\AppData\Local\anaconda3\envs\fws-dell\Lib\site-packages\django\template\base.py", line 1000, in render
    return SafeString("".join([node.render_annotated(context) for node in self]))
                               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Zhong_ming_Tan\AppData\Local\anaconda3\envs\fws-dell\Lib\site-packages\django\template\base.py", line 961, in render_annotated
    return self.render(context)
           ^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Zhong_ming_Tan\AppData\Local\anaconda3\envs\fws-dell\Lib\site-packages\django\template\defaulttags.py", line 479, in render
    url = reverse(view_name, args=args, kwargs=kwargs, current_app=current_app)
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Zhong_ming_Tan\AppData\Local\anaconda3\envs\fws-dell\Lib\site-packages\django\urls\base.py", line 88, in reverse
    return resolver._reverse_with_prefix(view, prefix, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Zhong_ming_Tan\AppData\Local\anaconda3\envs\fws-dell\Lib\site-packages\django\urls\resolvers.py", line 851, in _reverse_with_prefix
    raise NoReverseMatch(msg)
django.urls.exceptions.NoReverseMatch: Reverse for 'bitcoin_data_daily/get_all' not found. 'bitcoin_data_daily/get_all' is not a valid view function or pattern name.
2024-02-27 16:16:12,929 - INFO - C:\Other Project\Ads\Bitcoin Scraping\scrapperE\scrap\urls.py changed, reloading.
2024-02-27 16:16:20,303 - INFO - C:\Other Project\Ads\Bitcoin Scraping\scrapperE\scrap\views.py changed, reloading.
2024-02-27 16:16:35,880 - INFO - C:\Other Project\Ads\Bitcoin Scraping\scrapperE\scrap\urls.py changed, reloading.
2024-02-27 16:16:48,075 - INFO - C:\Other Project\Ads\Bitcoin Scraping\scrapperE\scrap\urls.py changed, reloading.
2024-02-27 16:17:02,769 - ERROR - Internal Server Error: /scrap/
Traceback (most recent call last):
  File "C:\Users\Zhong_ming_Tan\AppData\Local\anaconda3\envs\fws-dell\Lib\site-packages\django\core\handlers\exception.py", line 55, in inner
    response = get_response(request)
               ^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Zhong_ming_Tan\AppData\Local\anaconda3\envs\fws-dell\Lib\site-packages\django\core\handlers\base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Other Project\Ads\Bitcoin Scraping\scrapperE\scrap\views.py", line 31, in dashboard_with_pivot
    return render(request, 'dashboard_with_pivot.html', {})
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Zhong_ming_Tan\AppData\Local\anaconda3\envs\fws-dell\Lib\site-packages\django\shortcuts.py", line 25, in render
    content = loader.render_to_string(template_name, context, request, using=using)
              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Zhong_ming_Tan\AppData\Local\anaconda3\envs\fws-dell\Lib\site-packages\django\template\loader.py", line 62, in render_to_string
    return template.render(context, request)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Zhong_ming_Tan\AppData\Local\anaconda3\envs\fws-dell\Lib\site-packages\django\template\backends\django.py", line 61, in render
    return self.template.render(context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Zhong_ming_Tan\AppData\Local\anaconda3\envs\fws-dell\Lib\site-packages\django\template\base.py", line 171, in render
    return self._render(context)
           ^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Zhong_ming_Tan\AppData\Local\anaconda3\envs\fws-dell\Lib\site-packages\django\template\base.py", line 163, in _render
    return self.nodelist.render(context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Zhong_ming_Tan\AppData\Local\anaconda3\envs\fws-dell\Lib\site-packages\django\template\base.py", line 1000, in render
    return SafeString("".join([node.render_annotated(context) for node in self]))
                               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Zhong_ming_Tan\AppData\Local\anaconda3\envs\fws-dell\Lib\site-packages\django\template\base.py", line 961, in render_annotated
    return self.render(context)
           ^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Zhong_ming_Tan\AppData\Local\anaconda3\envs\fws-dell\Lib\site-packages\django\template\defaulttags.py", line 479, in render
    url = reverse(view_name, args=args, kwargs=kwargs, current_app=current_app)
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Zhong_ming_Tan\AppData\Local\anaconda3\envs\fws-dell\Lib\site-packages\django\urls\base.py", line 88, in reverse
    return resolver._reverse_with_prefix(view, prefix, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Zhong_ming_Tan\AppData\Local\anaconda3\envs\fws-dell\Lib\site-packages\django\urls\resolvers.py", line 851, in _reverse_with_prefix
    raise NoReverseMatch(msg)
django.urls.exceptions.NoReverseMatch: Reverse for 'all_post_raw' not found. 'all_post_raw' is not a valid view function or pattern name.
2024-02-27 16:17:22,346 - ERROR - Internal Server Error: /scrap/
Traceback (most recent call last):
  File "C:\Users\Zhong_ming_Tan\AppData\Local\anaconda3\envs\fws-dell\Lib\site-packages\django\core\handlers\exception.py", line 55, in inner
    response = get_response(request)
               ^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Zhong_ming_Tan\AppData\Local\anaconda3\envs\fws-dell\Lib\site-packages\django\core\handlers\base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Other Project\Ads\Bitcoin Scraping\scrapperE\scrap\views.py", line 31, in dashboard_with_pivot
    return render(request, 'dashboard_with_pivot.html', {})
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Zhong_ming_Tan\AppData\Local\anaconda3\envs\fws-dell\Lib\site-packages\django\shortcuts.py", line 25, in render
    content = loader.render_to_string(template_name, context, request, using=using)
              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Zhong_ming_Tan\AppData\Local\anaconda3\envs\fws-dell\Lib\site-packages\django\template\loader.py", line 62, in render_to_string
    return template.render(context, request)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Zhong_ming_Tan\AppData\Local\anaconda3\envs\fws-dell\Lib\site-packages\django\template\backends\django.py", line 61, in render
    return self.template.render(context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Zhong_ming_Tan\AppData\Local\anaconda3\envs\fws-dell\Lib\site-packages\django\template\base.py", line 171, in render
    return self._render(context)
           ^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Zhong_ming_Tan\AppData\Local\anaconda3\envs\fws-dell\Lib\site-packages\django\template\base.py", line 163, in _render
    return self.nodelist.render(context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Zhong_ming_Tan\AppData\Local\anaconda3\envs\fws-dell\Lib\site-packages\django\template\base.py", line 1000, in render
    return SafeString("".join([node.render_annotated(context) for node in self]))
                               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Zhong_ming_Tan\AppData\Local\anaconda3\envs\fws-dell\Lib\site-packages\django\template\base.py", line 961, in render_annotated
    return self.render(context)
           ^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Zhong_ming_Tan\AppData\Local\anaconda3\envs\fws-dell\Lib\site-packages\django\template\defaulttags.py", line 479, in render
    url = reverse(view_name, args=args, kwargs=kwargs, current_app=current_app)
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Zhong_ming_Tan\AppData\Local\anaconda3\envs\fws-dell\Lib\site-packages\django\urls\base.py", line 88, in reverse
    return resolver._reverse_with_prefix(view, prefix, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Zhong_ming_Tan\AppData\Local\anaconda3\envs\fws-dell\Lib\site-packages\django\urls\resolvers.py", line 851, in _reverse_with_prefix
    raise NoReverseMatch(msg)
django.urls.exceptions.NoReverseMatch: Reverse for 'all_post_raw' not found. 'all_post_raw' is not a valid view function or pattern name.
2024-02-27 16:17:24,917 - ERROR - Internal Server Error: /scrap/
Traceback (most recent call last):
  File "C:\Users\Zhong_ming_Tan\AppData\Local\anaconda3\envs\fws-dell\Lib\site-packages\django\core\handlers\exception.py", line 55, in inner
    response = get_response(request)
               ^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Zhong_ming_Tan\AppData\Local\anaconda3\envs\fws-dell\Lib\site-packages\django\core\handlers\base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Other Project\Ads\Bitcoin Scraping\scrapperE\scrap\views.py", line 31, in dashboard_with_pivot
    return render(request, 'dashboard_with_pivot.html', {})
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Zhong_ming_Tan\AppData\Local\anaconda3\envs\fws-dell\Lib\site-packages\django\shortcuts.py", line 25, in render
    content = loader.render_to_string(template_name, context, request, using=using)
              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Zhong_ming_Tan\AppData\Local\anaconda3\envs\fws-dell\Lib\site-packages\django\template\loader.py", line 62, in render_to_string
    return template.render(context, request)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Zhong_ming_Tan\AppData\Local\anaconda3\envs\fws-dell\Lib\site-packages\django\template\backends\django.py", line 61, in render
    return self.template.render(context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Zhong_ming_Tan\AppData\Local\anaconda3\envs\fws-dell\Lib\site-packages\django\template\base.py", line 171, in render
    return self._render(context)
           ^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Zhong_ming_Tan\AppData\Local\anaconda3\envs\fws-dell\Lib\site-packages\django\template\base.py", line 163, in _render
    return self.nodelist.render(context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Zhong_ming_Tan\AppData\Local\anaconda3\envs\fws-dell\Lib\site-packages\django\template\base.py", line 1000, in render
    return SafeString("".join([node.render_annotated(context) for node in self]))
                               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Zhong_ming_Tan\AppData\Local\anaconda3\envs\fws-dell\Lib\site-packages\django\template\base.py", line 961, in render_annotated
    return self.render(context)
           ^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Zhong_ming_Tan\AppData\Local\anaconda3\envs\fws-dell\Lib\site-packages\django\template\defaulttags.py", line 479, in render
    url = reverse(view_name, args=args, kwargs=kwargs, current_app=current_app)
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Zhong_ming_Tan\AppData\Local\anaconda3\envs\fws-dell\Lib\site-packages\django\urls\base.py", line 88, in reverse
    return resolver._reverse_with_prefix(view, prefix, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Zhong_ming_Tan\AppData\Local\anaconda3\envs\fws-dell\Lib\site-packages\django\urls\resolvers.py", line 851, in _reverse_with_prefix
    raise NoReverseMatch(msg)
django.urls.exceptions.NoReverseMatch: Reverse for 'all_post_raw' not found. 'all_post_raw' is not a valid view function or pattern name.
2024-02-27 16:17:36,273 - ERROR - Internal Server Error: /scrap/
Traceback (most recent call last):
  File "C:\Users\Zhong_ming_Tan\AppData\Local\anaconda3\envs\fws-dell\Lib\site-packages\django\core\handlers\exception.py", line 55, in inner
    response = get_response(request)
               ^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Zhong_ming_Tan\AppData\Local\anaconda3\envs\fws-dell\Lib\site-packages\django\core\handlers\base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Other Project\Ads\Bitcoin Scraping\scrapperE\scrap\views.py", line 31, in dashboard_with_pivot
    return render(request, 'dashboard_with_pivot.html', {})
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Zhong_ming_Tan\AppData\Local\anaconda3\envs\fws-dell\Lib\site-packages\django\shortcuts.py", line 25, in render
    content = loader.render_to_string(template_name, context, request, using=using)
              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Zhong_ming_Tan\AppData\Local\anaconda3\envs\fws-dell\Lib\site-packages\django\template\loader.py", line 62, in render_to_string
    return template.render(context, request)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Zhong_ming_Tan\AppData\Local\anaconda3\envs\fws-dell\Lib\site-packages\django\template\backends\django.py", line 61, in render
    return self.template.render(context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Zhong_ming_Tan\AppData\Local\anaconda3\envs\fws-dell\Lib\site-packages\django\template\base.py", line 171, in render
    return self._render(context)
           ^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Zhong_ming_Tan\AppData\Local\anaconda3\envs\fws-dell\Lib\site-packages\django\template\base.py", line 163, in _render
    return self.nodelist.render(context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Zhong_ming_Tan\AppData\Local\anaconda3\envs\fws-dell\Lib\site-packages\django\template\base.py", line 1000, in render
    return SafeString("".join([node.render_annotated(context) for node in self]))
                               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Zhong_ming_Tan\AppData\Local\anaconda3\envs\fws-dell\Lib\site-packages\django\template\base.py", line 961, in render_annotated
    return self.render(context)
           ^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Zhong_ming_Tan\AppData\Local\anaconda3\envs\fws-dell\Lib\site-packages\django\template\defaulttags.py", line 479, in render
    url = reverse(view_name, args=args, kwargs=kwargs, current_app=current_app)
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Zhong_ming_Tan\AppData\Local\anaconda3\envs\fws-dell\Lib\site-packages\django\urls\base.py", line 88, in reverse
    return resolver._reverse_with_prefix(view, prefix, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Zhong_ming_Tan\AppData\Local\anaconda3\envs\fws-dell\Lib\site-packages\django\urls\resolvers.py", line 851, in _reverse_with_prefix
    raise NoReverseMatch(msg)
django.urls.exceptions.NoReverseMatch: Reverse for 'data' not found. 'data' is not a valid view function or pattern name.
2024-02-27 16:17:59,736 - INFO - C:\Other Project\Ads\Bitcoin Scraping\scrapperE\scrap\urls.py changed, reloading.
2024-02-27 16:18:03,578 - INFO - C:\Other Project\Ads\Bitcoin Scraping\scrapperE\scrap\views.py changed, reloading.
2024-02-27 16:18:29,223 - INFO - C:\Other Project\Ads\Bitcoin Scraping\scrapperE\scrap\urls.py changed, reloading.
2024-02-27 16:18:40,197 - INFO - C:\Other Project\Ads\Bitcoin Scraping\scrapperE\scrap\urls.py changed, reloading.
2024-02-27 16:19:30,200 - INFO - C:\Other Project\Ads\Bitcoin Scraping\scrapperE\scrap\urls.py changed, reloading.
2024-02-27 16:19:33,905 - INFO - C:\Other Project\Ads\Bitcoin Scraping\scrapperE\scrap\urls.py changed, reloading.
2024-02-27 16:19:36,936 - ERROR - Internal Server Error: /scrap/
Traceback (most recent call last):
  File "C:\Users\Zhong_ming_Tan\AppData\Local\anaconda3\envs\fws-dell\Lib\site-packages\django\core\handlers\exception.py", line 55, in inner
    response = get_response(request)
               ^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Zhong_ming_Tan\AppData\Local\anaconda3\envs\fws-dell\Lib\site-packages\django\core\handlers\base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Other Project\Ads\Bitcoin Scraping\scrapperE\scrap\views.py", line 31, in dashboard_with_pivot
    return render(request, 'dashboard_with_pivot.html', {})
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Zhong_ming_Tan\AppData\Local\anaconda3\envs\fws-dell\Lib\site-packages\django\shortcuts.py", line 25, in render
    content = loader.render_to_string(template_name, context, request, using=using)
              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Zhong_ming_Tan\AppData\Local\anaconda3\envs\fws-dell\Lib\site-packages\django\template\loader.py", line 62, in render_to_string
    return template.render(context, request)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Zhong_ming_Tan\AppData\Local\anaconda3\envs\fws-dell\Lib\site-packages\django\template\backends\django.py", line 61, in render
    return self.template.render(context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Zhong_ming_Tan\AppData\Local\anaconda3\envs\fws-dell\Lib\site-packages\django\template\base.py", line 171, in render
    return self._render(context)
           ^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Zhong_ming_Tan\AppData\Local\anaconda3\envs\fws-dell\Lib\site-packages\django\template\base.py", line 163, in _render
    return self.nodelist.render(context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Zhong_ming_Tan\AppData\Local\anaconda3\envs\fws-dell\Lib\site-packages\django\template\base.py", line 1000, in render
    return SafeString("".join([node.render_annotated(context) for node in self]))
                               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Zhong_ming_Tan\AppData\Local\anaconda3\envs\fws-dell\Lib\site-packages\django\template\base.py", line 961, in render_annotated
    return self.render(context)
           ^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Zhong_ming_Tan\AppData\Local\anaconda3\envs\fws-dell\Lib\site-packages\django\template\defaulttags.py", line 479, in render
    url = reverse(view_name, args=args, kwargs=kwargs, current_app=current_app)
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Zhong_ming_Tan\AppData\Local\anaconda3\envs\fws-dell\Lib\site-packages\django\urls\base.py", line 88, in reverse
    return resolver._reverse_with_prefix(view, prefix, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Zhong_ming_Tan\AppData\Local\anaconda3\envs\fws-dell\Lib\site-packages\django\urls\resolvers.py", line 851, in _reverse_with_prefix
    raise NoReverseMatch(msg)
django.urls.exceptions.NoReverseMatch: Reverse for 'pivot_data' not found. 'pivot_data' is not a valid view function or pattern name.
2024-02-27 16:19:55,585 - INFO - C:\Other Project\Ads\Bitcoin Scraping\scrapperE\scrap\urls.py changed, reloading.
2024-02-27 16:19:59,687 - INFO - C:\Other Project\Ads\Bitcoin Scraping\scrapperE\scrap\views.py changed, reloading.
2024-02-27 16:20:42,480 - INFO - C:\Other Project\Ads\Bitcoin Scraping\scrapperE\scrap\urls.py changed, reloading.
2024-02-27 16:25:12,469 - INFO - C:\Other Project\Ads\Bitcoin Scraping\scrapperE\scrap\views.py changed, reloading.
2024-02-27 16:25:16,301 - INFO - C:\Other Project\Ads\Bitcoin Scraping\scrapperE\scrap\views.py changed, reloading.
2024-02-27 16:25:31,952 - INFO - C:\Other Project\Ads\Bitcoin Scraping\scrapperE\scrap\urls.py changed, reloading.
2024-02-27 16:25:35,447 - INFO - C:\Other Project\Ads\Bitcoin Scraping\scrapperE\scrap\urls.py changed, reloading.
2024-02-27 17:16:01,482 - ERROR - Internal Server Error: /scrap/
Traceback (most recent call last):
  File "C:\Users\Zhong_ming_Tan\AppData\Local\anaconda3\envs\fws-dell\Lib\site-packages\django\core\handlers\exception.py", line 55, in inner
    response = get_response(request)
               ^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Zhong_ming_Tan\AppData\Local\anaconda3\envs\fws-dell\Lib\site-packages\django\core\handlers\base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Other Project\Ads\Bitcoin Scraping\scrapperE\scrap\views.py", line 34, in dashboard_d3js
    return render(request, 'dashboard_d3js.html', {})
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Zhong_ming_Tan\AppData\Local\anaconda3\envs\fws-dell\Lib\site-packages\django\shortcuts.py", line 25, in render
    content = loader.render_to_string(template_name, context, request, using=using)
              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Zhong_ming_Tan\AppData\Local\anaconda3\envs\fws-dell\Lib\site-packages\django\template\loader.py", line 62, in render_to_string
    return template.render(context, request)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Zhong_ming_Tan\AppData\Local\anaconda3\envs\fws-dell\Lib\site-packages\django\template\backends\django.py", line 61, in render
    return self.template.render(context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Zhong_ming_Tan\AppData\Local\anaconda3\envs\fws-dell\Lib\site-packages\django\template\base.py", line 171, in render
    return self._render(context)
           ^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Zhong_ming_Tan\AppData\Local\anaconda3\envs\fws-dell\Lib\site-packages\django\template\base.py", line 163, in _render
    return self.nodelist.render(context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Zhong_ming_Tan\AppData\Local\anaconda3\envs\fws-dell\Lib\site-packages\django\template\base.py", line 1000, in render
    return SafeString("".join([node.render_annotated(context) for node in self]))
                               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Zhong_ming_Tan\AppData\Local\anaconda3\envs\fws-dell\Lib\site-packages\django\template\base.py", line 961, in render_annotated
    return self.render(context)
           ^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Zhong_ming_Tan\AppData\Local\anaconda3\envs\fws-dell\Lib\site-packages\django\template\defaulttags.py", line 479, in render
    url = reverse(view_name, args=args, kwargs=kwargs, current_app=current_app)
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Zhong_ming_Tan\AppData\Local\anaconda3\envs\fws-dell\Lib\site-packages\django\urls\base.py", line 88, in reverse
    return resolver._reverse_with_prefix(view, prefix, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Zhong_ming_Tan\AppData\Local\anaconda3\envs\fws-dell\Lib\site-packages\django\urls\resolvers.py", line 851, in _reverse_with_prefix
    raise NoReverseMatch(msg)
django.urls.exceptions.NoReverseMatch: Reverse for 'play_count_by_month' not found. 'play_count_by_month' is not a valid view function or pattern name.
2024-02-27 17:16:04,194 - ERROR - Internal Server Error: /scrap/
Traceback (most recent call last):
  File "C:\Users\Zhong_ming_Tan\AppData\Local\anaconda3\envs\fws-dell\Lib\site-packages\django\core\handlers\exception.py", line 55, in inner
    response = get_response(request)
               ^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Zhong_ming_Tan\AppData\Local\anaconda3\envs\fws-dell\Lib\site-packages\django\core\handlers\base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Other Project\Ads\Bitcoin Scraping\scrapperE\scrap\views.py", line 34, in dashboard_d3js
    return render(request, 'dashboard_d3js.html', {})
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Zhong_ming_Tan\AppData\Local\anaconda3\envs\fws-dell\Lib\site-packages\django\shortcuts.py", line 25, in render
    content = loader.render_to_string(template_name, context, request, using=using)
              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Zhong_ming_Tan\AppData\Local\anaconda3\envs\fws-dell\Lib\site-packages\django\template\loader.py", line 62, in render_to_string
    return template.render(context, request)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Zhong_ming_Tan\AppData\Local\anaconda3\envs\fws-dell\Lib\site-packages\django\template\backends\django.py", line 61, in render
    return self.template.render(context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Zhong_ming_Tan\AppData\Local\anaconda3\envs\fws-dell\Lib\site-packages\django\template\base.py", line 171, in render
    return self._render(context)
           ^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Zhong_ming_Tan\AppData\Local\anaconda3\envs\fws-dell\Lib\site-packages\django\template\base.py", line 163, in _render
    return self.nodelist.render(context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Zhong_ming_Tan\AppData\Local\anaconda3\envs\fws-dell\Lib\site-packages\django\template\base.py", line 1000, in render
    return SafeString("".join([node.render_annotated(context) for node in self]))
                               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Zhong_ming_Tan\AppData\Local\anaconda3\envs\fws-dell\Lib\site-packages\django\template\base.py", line 961, in render_annotated
    return self.render(context)
           ^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Zhong_ming_Tan\AppData\Local\anaconda3\envs\fws-dell\Lib\site-packages\django\template\defaulttags.py", line 479, in render
    url = reverse(view_name, args=args, kwargs=kwargs, current_app=current_app)
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Zhong_ming_Tan\AppData\Local\anaconda3\envs\fws-dell\Lib\site-packages\django\urls\base.py", line 88, in reverse
    return resolver._reverse_with_prefix(view, prefix, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Zhong_ming_Tan\AppData\Local\anaconda3\envs\fws-dell\Lib\site-packages\django\urls\resolvers.py", line 851, in _reverse_with_prefix
    raise NoReverseMatch(msg)
django.urls.exceptions.NoReverseMatch: Reverse for 'play_count_by_month' not found. 'play_count_by_month' is not a valid view function or pattern name.
2024-02-27 17:19:47,863 - INFO - C:\Other Project\Ads\Bitcoin Scraping\scrapperE\scrap\views.py changed, reloading.
2024-02-27 17:19:57,043 - ERROR - Internal Server Error: /scrap/bitcoin_hourly_data
Traceback (most recent call last):
  File "C:\Users\Zhong_ming_Tan\AppData\Local\anaconda3\envs\fws-dell\Lib\site-packages\django\core\handlers\exception.py", line 55, in inner
    response = get_response(request)
               ^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Zhong_ming_Tan\AppData\Local\anaconda3\envs\fws-dell\Lib\site-packages\django\utils\deprecation.py", line 136, in __call__
    response = self.process_response(request, response)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Zhong_ming_Tan\AppData\Local\anaconda3\envs\fws-dell\Lib\site-packages\django\middleware\clickjacking.py", line 27, in process_response
    if response.get("X-Frame-Options") is not None:
       ^^^^^^^^^^^^
AttributeError: 'str' object has no attribute 'get'
2024-02-27 17:20:14,851 - ERROR - Internal Server Error: /scrap/bitcoin_hourly_data
Traceback (most recent call last):
  File "C:\Users\Zhong_ming_Tan\AppData\Local\anaconda3\envs\fws-dell\Lib\site-packages\django\core\handlers\exception.py", line 55, in inner
    response = get_response(request)
               ^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Zhong_ming_Tan\AppData\Local\anaconda3\envs\fws-dell\Lib\site-packages\django\utils\deprecation.py", line 136, in __call__
    response = self.process_response(request, response)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Zhong_ming_Tan\AppData\Local\anaconda3\envs\fws-dell\Lib\site-packages\django\middleware\clickjacking.py", line 27, in process_response
    if response.get("X-Frame-Options") is not None:
       ^^^^^^^^^^^^
AttributeError: 'str' object has no attribute 'get'
2024-02-27 17:20:21,363 - INFO - C:\Other Project\Ads\Bitcoin Scraping\scrapperE\scrap\views.py changed, reloading.
2024-02-27 17:27:26,842 - INFO - C:\Other Project\Ads\Bitcoin Scraping\scrapperE\scrap\views.py changed, reloading.
2024-02-27 17:34:05,469 - INFO - C:\Other Project\Ads\Bitcoin Scraping\scrapperE\scrap\views.py changed, reloading.
2024-02-27 17:34:35,987 - INFO - C:\Other Project\Ads\Bitcoin Scraping\scrapperE\scrap\views.py changed, reloading.
2024-02-27 17:34:50,017 - INFO - C:\Other Project\Ads\Bitcoin Scraping\scrapperE\scrap\views.py changed, reloading.
2024-02-27 17:35:28,459 - INFO - C:\Other Project\Ads\Bitcoin Scraping\scrapperE\scrap\views.py changed, reloading.
2024-02-27 17:35:40,963 - INFO - C:\Other Project\Ads\Bitcoin Scraping\scrapperE\scrap\views.py changed, reloading.
2024-02-27 17:36:07,516 - INFO - C:\Other Project\Ads\Bitcoin Scraping\scrapperE\scrap\urls.py changed, reloading.
2024-02-27 17:36:14,387 - INFO - C:\Other Project\Ads\Bitcoin Scraping\scrapperE\scrap\urls.py changed, reloading.
2024-02-27 17:36:36,636 - ERROR - Internal Server Error: /scrap/bitstamphour_count_by_month
Traceback (most recent call last):
  File "C:\Users\Zhong_ming_Tan\AppData\Local\anaconda3\envs\fws-dell\Lib\site-packages\django\core\handlers\exception.py", line 55, in inner
    response = get_response(request)
               ^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Zhong_ming_Tan\AppData\Local\anaconda3\envs\fws-dell\Lib\site-packages\django\core\handlers\base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Other Project\Ads\Bitcoin Scraping\scrapperE\scrap\views.py", line 53, in bitstamphour_count_by_month
    'month': connections[BitstampDataHour.objects.db].ops.date_trunc_sql('month', 'local_date')
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
TypeError: DatabaseOperations.date_trunc_sql() missing 1 required positional argument: 'params'
2024-02-27 17:36:40,834 - ERROR - Internal Server Error: /scrap/bitstamphour_count_by_month
Traceback (most recent call last):
  File "C:\Users\Zhong_ming_Tan\AppData\Local\anaconda3\envs\fws-dell\Lib\site-packages\django\core\handlers\exception.py", line 55, in inner
    response = get_response(request)
               ^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Zhong_ming_Tan\AppData\Local\anaconda3\envs\fws-dell\Lib\site-packages\django\core\handlers\base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Other Project\Ads\Bitcoin Scraping\scrapperE\scrap\views.py", line 53, in bitstamphour_count_by_month
    'month': connections[BitstampDataHour.objects.db].ops.date_trunc_sql('month', 'local_date')
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
TypeError: DatabaseOperations.date_trunc_sql() missing 1 required positional argument: 'params'
2024-02-27 17:36:52,881 - INFO - C:\Other Project\Ads\Bitcoin Scraping\scrapperE\scrap\views.py changed, reloading.
2024-02-27 17:37:09,921 - INFO - C:\Other Project\Ads\Bitcoin Scraping\scrapperE\scrap\urls.py changed, reloading.
2024-02-27 17:37:17,599 - ERROR - Internal Server Error: /scrap/bitstamphour_count_by_month
Traceback (most recent call last):
  File "C:\Users\Zhong_ming_Tan\AppData\Local\anaconda3\envs\fws-dell\Lib\site-packages\django\core\handlers\exception.py", line 55, in inner
    response = get_response(request)
               ^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Zhong_ming_Tan\AppData\Local\anaconda3\envs\fws-dell\Lib\site-packages\django\core\handlers\base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Other Project\Ads\Bitcoin Scraping\scrapperE\scrap\views.py", line 53, in bitstamphour_count_by_month
    'month': connections[BitstampDataHour.objects.db].ops.date_trunc_sql('month', 'local_date')
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
TypeError: DatabaseOperations.date_trunc_sql() missing 1 required positional argument: 'params'
2024-02-27 17:38:48,065 - INFO - C:\Other Project\Ads\Bitcoin Scraping\scrapperE\scrap\views.py changed, reloading.
2024-02-27 17:38:59,336 - INFO - C:\Other Project\Ads\Bitcoin Scraping\scrapperE\scrap\views.py changed, reloading.
2024-02-27 17:39:04,474 - INFO - C:\Other Project\Ads\Bitcoin Scraping\scrapperE\scrap\views.py changed, reloading.
2024-02-27 17:39:20,177 - ERROR - Internal Server Error: /scrap/bitstamphour_count_by_month
Traceback (most recent call last):
  File "C:\Users\Zhong_ming_Tan\AppData\Local\anaconda3\envs\fws-dell\Lib\site-packages\django\core\handlers\exception.py", line 55, in inner
    response = get_response(request)
               ^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Zhong_ming_Tan\AppData\Local\anaconda3\envs\fws-dell\Lib\site-packages\django\core\handlers\base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Other Project\Ads\Bitcoin Scraping\scrapperE\scrap\views.py", line 53, in bitstamphour_count_by_month
    'month': connections[BitstampDataHour.objects.db].ops.date_trunc_sql('month', 'local_date')
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
TypeError: DatabaseOperations.date_trunc_sql() missing 1 required positional argument: 'params'
2024-02-27 17:39:51,862 - INFO - C:\Other Project\Ads\Bitcoin Scraping\scrapperE\scrap\views.py changed, reloading.
2024-02-27 17:39:54,794 - ERROR - Internal Server Error: /scrap/bitstamphour_count_by_month
Traceback (most recent call last):
  File "C:\Users\Zhong_ming_Tan\AppData\Local\anaconda3\envs\fws-dell\Lib\site-packages\django\core\handlers\exception.py", line 55, in inner
    response = get_response(request)
               ^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Zhong_ming_Tan\AppData\Local\anaconda3\envs\fws-dell\Lib\site-packages\django\core\handlers\base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Other Project\Ads\Bitcoin Scraping\scrapperE\scrap\views.py", line 51, in bitstamphour_count_by_month
    .extra(
     ^^^^^^
  File "C:\Users\Zhong_ming_Tan\AppData\Local\anaconda3\envs\fws-dell\Lib\site-packages\django\db\models\query.py", line 1731, in extra
    clone.query.add_extra(select, select_params, where, params, tables, order_by)
  File "C:\Users\Zhong_ming_Tan\AppData\Local\anaconda3\envs\fws-dell\Lib\site-packages\django\db\models\sql\query.py", line 2355, in add_extra
    entry_params.append(next(param_iter))
                        ^^^^^^^^^^^^^^^^
StopIteration
2024-02-27 17:39:59,050 - INFO - C:\Other Project\Ads\Bitcoin Scraping\scrapperE\scrap\views.py changed, reloading.
2024-02-27 17:40:02,759 - ERROR - Internal Server Error: /scrap/bitstamphour_count_by_month
Traceback (most recent call last):
  File "C:\Users\Zhong_ming_Tan\AppData\Local\anaconda3\envs\fws-dell\Lib\site-packages\django\core\handlers\exception.py", line 55, in inner
    response = get_response(request)
               ^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Zhong_ming_Tan\AppData\Local\anaconda3\envs\fws-dell\Lib\site-packages\django\core\handlers\base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Other Project\Ads\Bitcoin Scraping\scrapperE\scrap\views.py", line 53, in bitstamphour_count_by_month
    'month': connections[BitstampDataHour.objects.db].ops.date_trunc_sql('month', 'local_date')
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
TypeError: DatabaseOperations.date_trunc_sql() missing 1 required positional argument: 'params'
2024-02-27 17:41:38,211 - INFO - C:\Other Project\Ads\Bitcoin Scraping\scrapperE\scrap\views.py changed, reloading.
2024-02-27 17:41:42,232 - ERROR - Internal Server Error: /scrap/bitstamphour_count_by_month
Traceback (most recent call last):
  File "C:\Users\Zhong_ming_Tan\AppData\Local\anaconda3\envs\fws-dell\Lib\site-packages\django\core\handlers\exception.py", line 55, in inner
    response = get_response(request)
               ^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Zhong_ming_Tan\AppData\Local\anaconda3\envs\fws-dell\Lib\site-packages\django\core\handlers\base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
TypeError: bitstamphour_count_by_month() takes 0 positional arguments but 1 was given
2024-02-27 17:41:59,211 - INFO - C:\Other Project\Ads\Bitcoin Scraping\scrapperE\scrap\views.py changed, reloading.
2024-02-27 17:42:01,639 - ERROR - Internal Server Error: /scrap/bitstamphour_count_by_month
Traceback (most recent call last):
  File "C:\Users\Zhong_ming_Tan\AppData\Local\anaconda3\envs\fws-dell\Lib\site-packages\django\core\handlers\exception.py", line 55, in inner
    response = get_response(request)
               ^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Zhong_ming_Tan\AppData\Local\anaconda3\envs\fws-dell\Lib\site-packages\django\core\handlers\base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Other Project\Ads\Bitcoin Scraping\scrapperE\scrap\views.py", line 53, in bitstamphour_count_by_month
    'month': connections[BitstampDataHour.objects.db].ops.date_trunc_sql('month', 'local_date')
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
TypeError: DatabaseOperations.date_trunc_sql() missing 1 required positional argument: 'params'
2024-02-27 17:42:02,332 - ERROR - Internal Server Error: /scrap/bitstamphour_count_by_month
Traceback (most recent call last):
  File "C:\Users\Zhong_ming_Tan\AppData\Local\anaconda3\envs\fws-dell\Lib\site-packages\django\core\handlers\exception.py", line 55, in inner
    response = get_response(request)
               ^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Zhong_ming_Tan\AppData\Local\anaconda3\envs\fws-dell\Lib\site-packages\django\core\handlers\base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Other Project\Ads\Bitcoin Scraping\scrapperE\scrap\views.py", line 53, in bitstamphour_count_by_month
    'month': connections[BitstampDataHour.objects.db].ops.date_trunc_sql('month', 'local_date')
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
TypeError: DatabaseOperations.date_trunc_sql() missing 1 required positional argument: 'params'
2024-02-27 17:42:02,785 - ERROR - Internal Server Error: /scrap/bitstamphour_count_by_month
Traceback (most recent call last):
  File "C:\Users\Zhong_ming_Tan\AppData\Local\anaconda3\envs\fws-dell\Lib\site-packages\django\core\handlers\exception.py", line 55, in inner
    response = get_response(request)
               ^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Zhong_ming_Tan\AppData\Local\anaconda3\envs\fws-dell\Lib\site-packages\django\core\handlers\base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Other Project\Ads\Bitcoin Scraping\scrapperE\scrap\views.py", line 53, in bitstamphour_count_by_month
    'month': connections[BitstampDataHour.objects.db].ops.date_trunc_sql('month', 'local_date')
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
TypeError: DatabaseOperations.date_trunc_sql() missing 1 required positional argument: 'params'
2024-02-27 17:42:02,993 - ERROR - Internal Server Error: /scrap/bitstamphour_count_by_month
Traceback (most recent call last):
  File "C:\Users\Zhong_ming_Tan\AppData\Local\anaconda3\envs\fws-dell\Lib\site-packages\django\core\handlers\exception.py", line 55, in inner
    response = get_response(request)
               ^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Zhong_ming_Tan\AppData\Local\anaconda3\envs\fws-dell\Lib\site-packages\django\core\handlers\base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Other Project\Ads\Bitcoin Scraping\scrapperE\scrap\views.py", line 53, in bitstamphour_count_by_month
    'month': connections[BitstampDataHour.objects.db].ops.date_trunc_sql('month', 'local_date')
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
TypeError: DatabaseOperations.date_trunc_sql() missing 1 required positional argument: 'params'
2024-02-27 17:42:03,353 - ERROR - Internal Server Error: /scrap/bitstamphour_count_by_month
Traceback (most recent call last):
  File "C:\Users\Zhong_ming_Tan\AppData\Local\anaconda3\envs\fws-dell\Lib\site-packages\django\core\handlers\exception.py", line 55, in inner
    response = get_response(request)
               ^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Zhong_ming_Tan\AppData\Local\anaconda3\envs\fws-dell\Lib\site-packages\django\core\handlers\base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Other Project\Ads\Bitcoin Scraping\scrapperE\scrap\views.py", line 53, in bitstamphour_count_by_month
    'month': connections[BitstampDataHour.objects.db].ops.date_trunc_sql('month', 'local_date')
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
TypeError: DatabaseOperations.date_trunc_sql() missing 1 required positional argument: 'params'
2024-02-27 17:42:03,659 - ERROR - Internal Server Error: /scrap/bitstamphour_count_by_month
Traceback (most recent call last):
  File "C:\Users\Zhong_ming_Tan\AppData\Local\anaconda3\envs\fws-dell\Lib\site-packages\django\core\handlers\exception.py", line 55, in inner
    response = get_response(request)
               ^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Zhong_ming_Tan\AppData\Local\anaconda3\envs\fws-dell\Lib\site-packages\django\core\handlers\base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Other Project\Ads\Bitcoin Scraping\scrapperE\scrap\views.py", line 53, in bitstamphour_count_by_month
    'month': connections[BitstampDataHour.objects.db].ops.date_trunc_sql('month', 'local_date')
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
TypeError: DatabaseOperations.date_trunc_sql() missing 1 required positional argument: 'params'
2024-02-27 17:42:03,867 - ERROR - Internal Server Error: /scrap/bitstamphour_count_by_month
Traceback (most recent call last):
  File "C:\Users\Zhong_ming_Tan\AppData\Local\anaconda3\envs\fws-dell\Lib\site-packages\django\core\handlers\exception.py", line 55, in inner
    response = get_response(request)
               ^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Zhong_ming_Tan\AppData\Local\anaconda3\envs\fws-dell\Lib\site-packages\django\core\handlers\base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Other Project\Ads\Bitcoin Scraping\scrapperE\scrap\views.py", line 53, in bitstamphour_count_by_month
    'month': connections[BitstampDataHour.objects.db].ops.date_trunc_sql('month', 'local_date')
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
TypeError: DatabaseOperations.date_trunc_sql() missing 1 required positional argument: 'params'
2024-02-27 17:42:04,197 - ERROR - Internal Server Error: /scrap/bitstamphour_count_by_month
Traceback (most recent call last):
  File "C:\Users\Zhong_ming_Tan\AppData\Local\anaconda3\envs\fws-dell\Lib\site-packages\django\core\handlers\exception.py", line 55, in inner
    response = get_response(request)
               ^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Zhong_ming_Tan\AppData\Local\anaconda3\envs\fws-dell\Lib\site-packages\django\core\handlers\base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Other Project\Ads\Bitcoin Scraping\scrapperE\scrap\views.py", line 53, in bitstamphour_count_by_month
    'month': connections[BitstampDataHour.objects.db].ops.date_trunc_sql('month', 'local_date')
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
TypeError: DatabaseOperations.date_trunc_sql() missing 1 required positional argument: 'params'
2024-02-27 17:43:22,569 - ERROR - Internal Server Error: /scrap/bitstamphour_count_by_month
Traceback (most recent call last):
  File "C:\Users\Zhong_ming_Tan\AppData\Local\anaconda3\envs\fws-dell\Lib\site-packages\django\core\handlers\exception.py", line 55, in inner
    response = get_response(request)
               ^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Zhong_ming_Tan\AppData\Local\anaconda3\envs\fws-dell\Lib\site-packages\django\core\handlers\base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Other Project\Ads\Bitcoin Scraping\scrapperE\scrap\views.py", line 53, in bitstamphour_count_by_month
    'month': connections[BitstampDataHour.objects.db].ops.date_trunc_sql('month', 'local_date')
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
TypeError: DatabaseOperations.date_trunc_sql() missing 1 required positional argument: 'params'
2024-02-27 17:43:35,789 - INFO - C:\Other Project\Ads\Bitcoin Scraping\scrapperE\scrap\views.py changed, reloading.
2024-02-27 17:43:44,083 - ERROR - Internal Server Error: /scrap/bitstamphour_count_by_month
Traceback (most recent call last):
  File "C:\Users\Zhong_ming_Tan\AppData\Local\anaconda3\envs\fws-dell\Lib\site-packages\django\core\handlers\exception.py", line 55, in inner
    response = get_response(request)
               ^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Zhong_ming_Tan\AppData\Local\anaconda3\envs\fws-dell\Lib\site-packages\django\core\handlers\base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Other Project\Ads\Bitcoin Scraping\scrapperE\scrap\views.py", line 53, in bitstamphour_count_by_month
    'month': connections[BitstampDataHour.objects.db].ops.date_trunc_sql('month', 'local_date')
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
TypeError: DatabaseOperations.date_trunc_sql() missing 1 required positional argument: 'params'
2024-02-27 17:54:32,806 - INFO - C:\Other Project\Ads\Bitcoin Scraping\scrapperE\scrap\views.py changed, reloading.
2024-02-27 17:54:38,768 - INFO - C:\Other Project\Ads\Bitcoin Scraping\scrapperE\scrap\urls.py changed, reloading.
2024-02-27 17:58:46,596 - INFO - C:\Other Project\Ads\Bitcoin Scraping\scrapperE\scrap\views.py changed, reloading.
2024-02-27 17:59:58,609 - INFO - C:\Other Project\Ads\Bitcoin Scraping\scrapperE\scrap\views.py changed, reloading.
2024-02-27 18:00:54,652 - ERROR - Internal Server Error: /scrap/
Traceback (most recent call last):
  File "C:\Users\Zhong_ming_Tan\AppData\Local\anaconda3\envs\fws-dell\Lib\site-packages\django\core\handlers\exception.py", line 55, in inner
    response = get_response(request)
               ^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Zhong_ming_Tan\AppData\Local\anaconda3\envs\fws-dell\Lib\site-packages\django\core\handlers\base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Other Project\Ads\Bitcoin Scraping\scrapperE\scrap\views.py", line 38, in dashboard_d3js
    return render(request, 'dashboard_d3js.html', {data})
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Zhong_ming_Tan\AppData\Local\anaconda3\envs\fws-dell\Lib\site-packages\django\shortcuts.py", line 25, in render
    content = loader.render_to_string(template_name, context, request, using=using)
              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Zhong_ming_Tan\AppData\Local\anaconda3\envs\fws-dell\Lib\site-packages\django\template\loader.py", line 62, in render_to_string
    return template.render(context, request)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Zhong_ming_Tan\AppData\Local\anaconda3\envs\fws-dell\Lib\site-packages\django\template\backends\django.py", line 57, in render
    context = make_context(
              ^^^^^^^^^^^^^
  File "C:\Users\Zhong_ming_Tan\AppData\Local\anaconda3\envs\fws-dell\Lib\site-packages\django\template\context.py", line 286, in make_context
    raise TypeError(
TypeError: context must be a dict rather than set.
2024-02-27 18:03:28,971 - ERROR - Internal Server Error: /scrap/
Traceback (most recent call last):
  File "C:\Users\Zhong_ming_Tan\AppData\Local\anaconda3\envs\fws-dell\Lib\site-packages\django\core\handlers\exception.py", line 55, in inner
    response = get_response(request)
               ^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Zhong_ming_Tan\AppData\Local\anaconda3\envs\fws-dell\Lib\site-packages\django\core\handlers\base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Other Project\Ads\Bitcoin Scraping\scrapperE\scrap\views.py", line 38, in dashboard_d3js
    return render(request, 'dashboard_d3js.html', {data})
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Zhong_ming_Tan\AppData\Local\anaconda3\envs\fws-dell\Lib\site-packages\django\shortcuts.py", line 25, in render
    content = loader.render_to_string(template_name, context, request, using=using)
              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Zhong_ming_Tan\AppData\Local\anaconda3\envs\fws-dell\Lib\site-packages\django\template\loader.py", line 62, in render_to_string
    return template.render(context, request)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Zhong_ming_Tan\AppData\Local\anaconda3\envs\fws-dell\Lib\site-packages\django\template\backends\django.py", line 57, in render
    context = make_context(
              ^^^^^^^^^^^^^
  File "C:\Users\Zhong_ming_Tan\AppData\Local\anaconda3\envs\fws-dell\Lib\site-packages\django\template\context.py", line 286, in make_context
    raise TypeError(
TypeError: context must be a dict rather than set.
2024-02-27 18:33:31,426 - ERROR - Internal Server Error: /scrap/
Traceback (most recent call last):
  File "C:\Users\Zhong_ming_Tan\AppData\Local\anaconda3\envs\fws-dell\Lib\site-packages\django\core\handlers\exception.py", line 55, in inner
    response = get_response(request)
               ^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Zhong_ming_Tan\AppData\Local\anaconda3\envs\fws-dell\Lib\site-packages\django\core\handlers\base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Other Project\Ads\Bitcoin Scraping\scrapperE\scrap\views.py", line 38, in dashboard_d3js
    return render(request, 'dashboard_d3js.html', {data})
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Zhong_ming_Tan\AppData\Local\anaconda3\envs\fws-dell\Lib\site-packages\django\shortcuts.py", line 25, in render
    content = loader.render_to_string(template_name, context, request, using=using)
              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Zhong_ming_Tan\AppData\Local\anaconda3\envs\fws-dell\Lib\site-packages\django\template\loader.py", line 62, in render_to_string
    return template.render(context, request)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Zhong_ming_Tan\AppData\Local\anaconda3\envs\fws-dell\Lib\site-packages\django\template\backends\django.py", line 57, in render
    context = make_context(
              ^^^^^^^^^^^^^
  File "C:\Users\Zhong_ming_Tan\AppData\Local\anaconda3\envs\fws-dell\Lib\site-packages\django\template\context.py", line 286, in make_context
    raise TypeError(
TypeError: context must be a dict rather than set.
2024-02-27 21:02:23,062 - INFO - C:\Other Project\Ads\Bitcoin Scraping\scrapperE\scrap\views.py changed, reloading.
2024-02-27 21:02:29,724 - WARNING - Not Found: /scrap/[object Object],[object Object],[object Object]
2024-02-27 21:02:38,762 - WARNING - Not Found: /scrap/[object Object],[object Object],[object Object]
2024-02-27 21:03:08,731 - WARNING - Not Found: /scrap/[object Object],[object Object],[object Object]
2024-02-27 21:03:11,342 - WARNING - Not Found: /scrap/[object Object],[object Object],[object Object]
2024-02-27 21:03:12,219 - WARNING - Not Found: /scrap/[object Object],[object Object],[object Object]
2024-02-27 21:03:43,832 - INFO - C:\Other Project\Ads\Bitcoin Scraping\scrapperE\scrap\views.py changed, reloading.
2024-02-27 21:03:50,859 - WARNING - Not Found: /scrap/[object Object],[object Object],[object Object]
2024-02-27 21:04:25,332 - INFO - C:\Other Project\Ads\Bitcoin Scraping\scrapperE\scrap\views.py changed, reloading.
2024-02-27 21:04:27,764 - WARNING - Not Found: /scrap/[object Object],[object Object],[object Object]
2024-02-27 21:04:30,663 - WARNING - Not Found: /scrap/[object Object],[object Object],[object Object]
2024-02-27 21:04:31,356 - WARNING - Not Found: /scrap/[object Object],[object Object],[object Object]
2024-02-27 21:04:31,945 - WARNING - Not Found: /scrap/[object Object],[object Object],[object Object]
2024-02-27 21:05:16,744 - WARNING - Not Found: /scrap/[object Object],[object Object],[object Object]
2024-02-27 21:05:18,771 - WARNING - Not Found: /scrap/[object Object],[object Object],[object Object]
2024-02-27 21:05:19,388 - WARNING - Not Found: /scrap/[object Object],[object Object],[object Object]
2024-02-27 21:05:20,087 - WARNING - Not Found: /scrap/[object Object],[object Object],[object Object]
2024-02-27 21:06:23,714 - INFO - C:\Other Project\Ads\Bitcoin Scraping\scrapperE\scrap\urls.py changed, reloading.
2024-02-27 21:06:41,437 - INFO - C:\Other Project\Ads\Bitcoin Scraping\scrapperE\scrap\views.py changed, reloading.
2024-02-27 21:06:43,940 - WARNING - Not Found: /scrap/
2024-02-27 21:06:44,510 - WARNING - Not Found: /scrap/bitcoin_hourly_data_close
2024-02-27 21:06:45,839 - WARNING - Not Found: /scrap/bitcoin_hourly_data_close
2024-02-27 21:06:47,928 - WARNING - Not Found: /scrap/
2024-02-27 21:09:17,928 - WARNING - Not Found: /scrap/
2024-02-27 21:09:18,628 - WARNING - Not Found: /scrap/
2024-02-27 21:09:19,295 - WARNING - Not Found: /scrap/
2024-02-27 21:09:44,881 - INFO - C:\Other Project\Ads\Bitcoin Scraping\scrapperE\scrap\urls.py changed, reloading.
2024-02-27 21:10:07,109 - INFO - C:\Other Project\Ads\Bitcoin Scraping\scrapperE\scrap\views.py changed, reloading.
2024-02-27 21:10:11,434 - WARNING - Not Found: /data/users.json
2024-02-27 21:10:17,680 - WARNING - Not Found: /data/users.json
2024-02-27 21:10:33,678 - WARNING - Not Found: /scrap/[object Object],[object Object]
2024-02-27 21:10:36,463 - WARNING - Not Found: /scrap/[object Object],[object Object]
2024-02-27 21:10:36,901 - WARNING - Not Found: /scrap/[object Object],[object Object]
2024-02-27 21:10:37,329 - WARNING - Not Found: /scrap/[object Object],[object Object]
2024-02-27 21:10:37,933 - WARNING - Not Found: /scrap/[object Object],[object Object]
2024-02-27 21:10:38,805 - WARNING - Not Found: /scrap/[object Object],[object Object]
2024-02-27 21:10:39,420 - WARNING - Not Found: /scrap/[object Object],[object Object]
2024-02-27 21:10:39,927 - WARNING - Not Found: /scrap/[object Object],[object Object]
2024-02-27 21:10:40,406 - WARNING - Not Found: /scrap/[object Object],[object Object]
2024-02-27 21:10:40,925 - WARNING - Not Found: /scrap/[object Object],[object Object]
2024-02-27 21:16:40,442 - WARNING - Not Found: /scrap/bitcoin_hourly_data.csv
2024-02-27 21:17:19,822 - WARNING - Not Found: /scrap/bitcoin_hourly_data.csv
2024-02-27 21:17:21,533 - WARNING - Not Found: /scrap/bitcoin_hourly_data.csv
2024-02-27 21:17:41,319 - INFO - C:\Other Project\Ads\Bitcoin Scraping\scrapperE\scrap\views.py changed, reloading.
2024-02-27 21:20:47,058 - INFO - C:\Other Project\Ads\Bitcoin Scraping\scrapperE\scrap\urls.py changed, reloading.
2024-02-27 21:21:02,798 - INFO - C:\Other Project\Ads\Bitcoin Scraping\scrapperE\scrap\views.py changed, reloading.
2024-02-27 21:21:51,680 - WARNING - Not Found: /scrap/
2024-02-27 21:25:04,284 - INFO - C:\Other Project\Ads\Bitcoin Scraping\scrapperE\scrap\views.py changed, reloading.
2024-02-27 21:25:13,906 - INFO - C:\Other Project\Ads\Bitcoin Scraping\scrapperE\scrap\views.py changed, reloading.
2024-02-27 21:25:25,769 - INFO - C:\Other Project\Ads\Bitcoin Scraping\scrapperE\scrap\urls.py changed, reloading.
2024-02-27 21:25:32,895 - ERROR - Internal Server Error: /scrap/sneat
Traceback (most recent call last):
  File "C:\Users\Zhong_ming_Tan\AppData\Local\anaconda3\envs\fws-dell\Lib\site-packages\django\core\handlers\exception.py", line 55, in inner
    response = get_response(request)
               ^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Zhong_ming_Tan\AppData\Local\anaconda3\envs\fws-dell\Lib\site-packages\django\core\handlers\base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Other Project\Ads\Bitcoin Scraping\scrapperE\scrap\views.py", line 37, in sneat
    return render(request, '/sneat-1.0.0/html/index.html', {})
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Zhong_ming_Tan\AppData\Local\anaconda3\envs\fws-dell\Lib\site-packages\django\shortcuts.py", line 25, in render
    content = loader.render_to_string(template_name, context, request, using=using)
              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Zhong_ming_Tan\AppData\Local\anaconda3\envs\fws-dell\Lib\site-packages\django\template\loader.py", line 61, in render_to_string
    template = get_template(template_name, using=using)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Zhong_ming_Tan\AppData\Local\anaconda3\envs\fws-dell\Lib\site-packages\django\template\loader.py", line 19, in get_template
    raise TemplateDoesNotExist(template_name, chain=chain)
django.template.exceptions.TemplateDoesNotExist: /sneat-1.0.0/html/index.html
2024-02-27 21:25:37,647 - INFO - C:\Other Project\Ads\Bitcoin Scraping\scrapperE\scrap\urls.py changed, reloading.
2024-02-27 21:25:41,833 - WARNING - Not Found: /scrap/sneat
2024-02-27 21:29:22,844 - INFO - C:\Other Project\Ads\Bitcoin Scraping\scrapperE\scrap\urls.py changed, reloading.
2024-02-27 21:29:31,415 - INFO - C:\Other Project\Ads\Bitcoin Scraping\scrapperE\scrap\views.py changed, reloading.
2024-02-27 21:29:34,994 - WARNING - Not Found: /scrap/sneat
2024-02-27 21:29:38,045 - WARNING - Not Found: /
2024-02-27 21:30:02,328 - INFO - C:\Other Project\Ads\Bitcoin Scraping\scrapperE\scrap\views.py changed, reloading.
2024-02-27 21:30:15,137 - INFO - C:\Other Project\Ads\Bitcoin Scraping\scrapperE\scrap\views.py changed, reloading.
2024-02-27 22:10:15,547 - WARNING - Not Found: /docs/5.3/assets/brand/bootstrap-logo.svg
2024-02-27 22:10:20,533 - WARNING - Not Found: /docs/5.3/assets/brand/bootstrap-logo.svg
2024-02-27 22:10:50,713 - WARNING - Not Found: /docs/5.3/assets/brand/bootstrap-logo.svg
2024-02-27 22:10:52,914 - WARNING - Not Found: /docs/5.3/assets/brand/bootstrap-logo.svg
2024-02-27 22:11:18,930 - WARNING - Not Found: /docs/5.3/assets/brand/bootstrap-logo.svg
2024-02-27 22:11:29,515 - WARNING - Not Found: /docs/5.3/assets/brand/bootstrap-logo.svg
2024-02-27 22:11:37,896 - WARNING - Not Found: /docs/5.3/assets/brand/bootstrap-logo.svg
2024-02-27 22:12:04,254 - WARNING - Not Found: /docs/5.3/assets/brand/bootstrap-logo.svg
2024-02-27 22:13:11,967 - WARNING - Not Found: /docs/5.3/assets/brand/bootstrap-logo.svg
2024-02-27 22:13:37,639 - WARNING - Not Found: /docs/5.3/assets/brand/bootstrap-logo.svg
2024-02-27 22:14:31,187 - WARNING - Not Found: /docs/5.3/assets/brand/bootstrap-logo.svg
2024-02-27 22:44:44,595 - INFO - C:\Other Project\Ads\Bitcoin Scraping\scrapperE\scrap\views.py changed, reloading.
2024-02-27 22:47:30,076 - INFO - C:\Other Project\Ads\Bitcoin Scraping\scrapperE\scrap\views.py changed, reloading.
2024-02-27 22:48:23,552 - INFO - C:\Other Project\Ads\Bitcoin Scraping\scrapperE\scrap\views.py changed, reloading.
2024-02-27 22:48:53,479 - INFO - C:\Other Project\Ads\Bitcoin Scraping\scrapperE\scrap\views.py changed, reloading.
2024-02-27 22:49:01,728 - INFO - C:\Other Project\Ads\Bitcoin Scraping\scrapperE\scrap\views.py changed, reloading.
2024-02-27 22:49:10,990 - INFO - C:\Other Project\Ads\Bitcoin Scraping\scrapperE\scrap\views.py changed, reloading.
2024-02-27 22:49:30,875 - INFO - C:\Other Project\Ads\Bitcoin Scraping\scrapperE\scrap\views.py changed, reloading.
2024-02-27 22:50:41,483 - INFO - C:\Other Project\Ads\Bitcoin Scraping\scrapperE\scrap\views.py changed, reloading.
2024-02-27 22:50:49,203 - INFO - C:\Other Project\Ads\Bitcoin Scraping\scrapperE\scrap\views.py changed, reloading.
2024-02-27 22:51:17,671 - INFO - C:\Other Project\Ads\Bitcoin Scraping\scrapperE\scrap\views.py changed, reloading.
2024-02-27 22:51:32,094 - INFO - C:\Other Project\Ads\Bitcoin Scraping\scrapperE\scrap\views.py changed, reloading.
2024-02-27 22:53:11,183 - INFO - C:\Other Project\Ads\Bitcoin Scraping\scrapperE\scrap\views.py changed, reloading.
2024-02-27 22:53:27,340 - INFO - C:\Other Project\Ads\Bitcoin Scraping\scrapperE\scrap\views.py changed, reloading.
2024-02-27 22:54:08,842 - INFO - C:\Other Project\Ads\Bitcoin Scraping\scrapperE\scrap\views.py changed, reloading.
2024-02-27 22:54:16,155 - INFO - C:\Other Project\Ads\Bitcoin Scraping\scrapperE\scrap\views.py changed, reloading.
2024-02-27 23:52:24,777 - WARNING - Not Found: /
2024-02-27 23:52:31,205 - ERROR - Internal Server Error: /register/
Traceback (most recent call last):
  File "C:\Users\Zhong_ming_Tan\AppData\Local\anaconda3\envs\fws-dell\Lib\site-packages\django\core\handlers\exception.py", line 55, in inner
    response = get_response(request)
               ^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Zhong_ming_Tan\AppData\Local\anaconda3\envs\fws-dell\Lib\site-packages\django\core\handlers\base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Other Project\Ads\Bitcoin Scraping\scrapperE\user\views.py", line 16, in register
    return render(request,'users/register.html',{'form':form})
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Zhong_ming_Tan\AppData\Local\anaconda3\envs\fws-dell\Lib\site-packages\django\shortcuts.py", line 25, in render
    content = loader.render_to_string(template_name, context, request, using=using)
              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Zhong_ming_Tan\AppData\Local\anaconda3\envs\fws-dell\Lib\site-packages\django\template\loader.py", line 61, in render_to_string
    template = get_template(template_name, using=using)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Zhong_ming_Tan\AppData\Local\anaconda3\envs\fws-dell\Lib\site-packages\django\template\loader.py", line 19, in get_template
    raise TemplateDoesNotExist(template_name, chain=chain)
django.template.exceptions.TemplateDoesNotExist: users/register.html
2024-02-27 23:52:57,459 - INFO - C:\Other Project\Ads\Bitcoin Scraping\scrapperE\scrapperE\urls.py changed, reloading.
2024-02-27 23:53:02,349 - ERROR - Internal Server Error: /register/
Traceback (most recent call last):
  File "C:\Users\Zhong_ming_Tan\AppData\Local\anaconda3\envs\fws-dell\Lib\site-packages\django\core\handlers\exception.py", line 55, in inner
    response = get_response(request)
               ^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Zhong_ming_Tan\AppData\Local\anaconda3\envs\fws-dell\Lib\site-packages\django\core\handlers\base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Other Project\Ads\Bitcoin Scraping\scrapperE\user\views.py", line 16, in register
    return render(request,'users/register.html',{'form':form})
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Zhong_ming_Tan\AppData\Local\anaconda3\envs\fws-dell\Lib\site-packages\django\shortcuts.py", line 25, in render
    content = loader.render_to_string(template_name, context, request, using=using)
              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Zhong_ming_Tan\AppData\Local\anaconda3\envs\fws-dell\Lib\site-packages\django\template\loader.py", line 61, in render_to_string
    template = get_template(template_name, using=using)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Zhong_ming_Tan\AppData\Local\anaconda3\envs\fws-dell\Lib\site-packages\django\template\loader.py", line 19, in get_template
    raise TemplateDoesNotExist(template_name, chain=chain)
django.template.exceptions.TemplateDoesNotExist: users/register.html
2024-02-27 23:53:06,077 - INFO - C:\Other Project\Ads\Bitcoin Scraping\scrapperE\scrapperE\urls.py changed, reloading.
2024-02-27 23:53:16,378 - ERROR - Internal Server Error: /register/
Traceback (most recent call last):
  File "C:\Users\Zhong_ming_Tan\AppData\Local\anaconda3\envs\fws-dell\Lib\site-packages\django\core\handlers\exception.py", line 55, in inner
    response = get_response(request)
               ^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Zhong_ming_Tan\AppData\Local\anaconda3\envs\fws-dell\Lib\site-packages\django\core\handlers\base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Other Project\Ads\Bitcoin Scraping\scrapperE\user\views.py", line 16, in register
    return render(request,'users/register.html',{'form':form})
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Zhong_ming_Tan\AppData\Local\anaconda3\envs\fws-dell\Lib\site-packages\django\shortcuts.py", line 25, in render
    content = loader.render_to_string(template_name, context, request, using=using)
              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Zhong_ming_Tan\AppData\Local\anaconda3\envs\fws-dell\Lib\site-packages\django\template\loader.py", line 61, in render_to_string
    template = get_template(template_name, using=using)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Zhong_ming_Tan\AppData\Local\anaconda3\envs\fws-dell\Lib\site-packages\django\template\loader.py", line 19, in get_template
    raise TemplateDoesNotExist(template_name, chain=chain)
django.template.exceptions.TemplateDoesNotExist: users/register.html
2024-02-27 23:53:17,271 - ERROR - Internal Server Error: /register/
Traceback (most recent call last):
  File "C:\Users\Zhong_ming_Tan\AppData\Local\anaconda3\envs\fws-dell\Lib\site-packages\django\core\handlers\exception.py", line 55, in inner
    response = get_response(request)
               ^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Zhong_ming_Tan\AppData\Local\anaconda3\envs\fws-dell\Lib\site-packages\django\core\handlers\base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Other Project\Ads\Bitcoin Scraping\scrapperE\user\views.py", line 16, in register
    return render(request,'users/register.html',{'form':form})
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Zhong_ming_Tan\AppData\Local\anaconda3\envs\fws-dell\Lib\site-packages\django\shortcuts.py", line 25, in render
    content = loader.render_to_string(template_name, context, request, using=using)
              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Zhong_ming_Tan\AppData\Local\anaconda3\envs\fws-dell\Lib\site-packages\django\template\loader.py", line 61, in render_to_string
    template = get_template(template_name, using=using)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Zhong_ming_Tan\AppData\Local\anaconda3\envs\fws-dell\Lib\site-packages\django\template\loader.py", line 19, in get_template
    raise TemplateDoesNotExist(template_name, chain=chain)
django.template.exceptions.TemplateDoesNotExist: users/register.html
2024-02-27 23:53:17,831 - ERROR - Internal Server Error: /register/
Traceback (most recent call last):
  File "C:\Users\Zhong_ming_Tan\AppData\Local\anaconda3\envs\fws-dell\Lib\site-packages\django\core\handlers\exception.py", line 55, in inner
    response = get_response(request)
               ^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Zhong_ming_Tan\AppData\Local\anaconda3\envs\fws-dell\Lib\site-packages\django\core\handlers\base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Other Project\Ads\Bitcoin Scraping\scrapperE\user\views.py", line 16, in register
    return render(request,'users/register.html',{'form':form})
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Zhong_ming_Tan\AppData\Local\anaconda3\envs\fws-dell\Lib\site-packages\django\shortcuts.py", line 25, in render
    content = loader.render_to_string(template_name, context, request, using=using)
              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Zhong_ming_Tan\AppData\Local\anaconda3\envs\fws-dell\Lib\site-packages\django\template\loader.py", line 61, in render_to_string
    template = get_template(template_name, using=using)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Zhong_ming_Tan\AppData\Local\anaconda3\envs\fws-dell\Lib\site-packages\django\template\loader.py", line 19, in get_template
    raise TemplateDoesNotExist(template_name, chain=chain)
django.template.exceptions.TemplateDoesNotExist: users/register.html
2024-02-27 23:54:21,694 - INFO - C:\Other Project\Ads\Bitcoin Scraping\scrapperE\user\views.py changed, reloading.
2024-02-27 23:56:36,293 - WARNING - Not Found: /accounts/profile/
2024-02-27 23:59:20,409 - INFO - C:\Other Project\Ads\Bitcoin Scraping\scrapperE\scrapperE\settings.py changed, reloading.
2024-02-27 23:59:28,085 - INFO - C:\Other Project\Ads\Bitcoin Scraping\scrapperE\scrapperE\settings.py changed, reloading.
2024-02-27 23:59:45,680 - ERROR - Internal Server Error: /login/
Traceback (most recent call last):
  File "C:\Users\Zhong_ming_Tan\AppData\Local\anaconda3\envs\fws-dell\Lib\site-packages\django\core\handlers\exception.py", line 55, in inner
    response = get_response(request)
               ^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Zhong_ming_Tan\AppData\Local\anaconda3\envs\fws-dell\Lib\site-packages\django\core\handlers\base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Zhong_ming_Tan\AppData\Local\anaconda3\envs\fws-dell\Lib\site-packages\django\views\generic\base.py", line 104, in view
    return self.dispatch(request, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Zhong_ming_Tan\AppData\Local\anaconda3\envs\fws-dell\Lib\site-packages\django\utils\decorators.py", line 48, in _wrapper
    return bound_method(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Zhong_ming_Tan\AppData\Local\anaconda3\envs\fws-dell\Lib\site-packages\django\views\decorators\debug.py", line 141, in sensitive_post_parameters_wrapper
    return view(request, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Zhong_ming_Tan\AppData\Local\anaconda3\envs\fws-dell\Lib\site-packages\django\utils\decorators.py", line 48, in _wrapper
    return bound_method(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Zhong_ming_Tan\AppData\Local\anaconda3\envs\fws-dell\Lib\site-packages\django\utils\decorators.py", line 188, in _view_wrapper
    result = _process_exception(request, e)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Zhong_ming_Tan\AppData\Local\anaconda3\envs\fws-dell\Lib\site-packages\django\utils\decorators.py", line 186, in _view_wrapper
    response = view_func(request, *args, **kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Zhong_ming_Tan\AppData\Local\anaconda3\envs\fws-dell\Lib\site-packages\django\utils\decorators.py", line 48, in _wrapper
    return bound_method(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Zhong_ming_Tan\AppData\Local\anaconda3\envs\fws-dell\Lib\site-packages\django\views\decorators\cache.py", line 80, in _view_wrapper
    response = view_func(request, *args, **kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Zhong_ming_Tan\AppData\Local\anaconda3\envs\fws-dell\Lib\site-packages\django\contrib\auth\views.py", line 88, in dispatch
    return super().dispatch(request, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Zhong_ming_Tan\AppData\Local\anaconda3\envs\fws-dell\Lib\site-packages\django\views\generic\base.py", line 143, in dispatch
    return handler(request, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Zhong_ming_Tan\AppData\Local\anaconda3\envs\fws-dell\Lib\site-packages\django\views\generic\edit.py", line 151, in post
    return self.form_valid(form)
           ^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Zhong_ming_Tan\AppData\Local\anaconda3\envs\fws-dell\Lib\site-packages\django\contrib\auth\views.py", line 108, in form_valid
    return HttpResponseRedirect(self.get_success_url())
                                ^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Zhong_ming_Tan\AppData\Local\anaconda3\envs\fws-dell\Lib\site-packages\django\contrib\auth\views.py", line 41, in get_success_url
    return self.get_redirect_url() or self.get_default_redirect_url()
                                      ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Zhong_ming_Tan\AppData\Local\anaconda3\envs\fws-dell\Lib\site-packages\django\contrib\auth\views.py", line 95, in get_default_redirect_url
    return resolve_url(settings.LOGIN_REDIRECT_URL)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Zhong_ming_Tan\AppData\Local\anaconda3\envs\fws-dell\Lib\site-packages\django\shortcuts.py", line 180, in resolve_url
    return reverse(to, args=args, kwargs=kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Zhong_ming_Tan\AppData\Local\anaconda3\envs\fws-dell\Lib\site-packages\django\urls\base.py", line 88, in reverse
    return resolver._reverse_with_prefix(view, prefix, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Zhong_ming_Tan\AppData\Local\anaconda3\envs\fws-dell\Lib\site-packages\django\urls\resolvers.py", line 851, in _reverse_with_prefix
    raise NoReverseMatch(msg)
django.urls.exceptions.NoReverseMatch: Reverse for 'scrap' not found. 'scrap' is not a valid view function or pattern name.
2024-02-28 00:00:11,261 - INFO - C:\Other Project\Ads\Bitcoin Scraping\scrapperE\scrapperE\settings.py changed, reloading.
2024-02-28 00:00:18,452 - WARNING - Forbidden (CSRF token from POST incorrect.): /login/
2024-02-28 00:00:20,143 - WARNING - Forbidden (CSRF token from POST incorrect.): /login/
2024-02-28 00:00:36,219 - WARNING - Not Found: /login/scrap/
2024-02-28 00:01:21,579 - INFO - C:\Other Project\Ads\Bitcoin Scraping\scrapperE\scrapperE\urls.py changed, reloading.
2024-02-28 00:01:35,004 - INFO - C:\Other Project\Ads\Bitcoin Scraping\scrapperE\scrapperE\settings.py changed, reloading.
2024-02-28 00:01:47,927 - ERROR - Internal Server Error: /login/
Traceback (most recent call last):
  File "C:\Users\Zhong_ming_Tan\AppData\Local\anaconda3\envs\fws-dell\Lib\site-packages\django\core\handlers\exception.py", line 55, in inner
    response = get_response(request)
               ^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Zhong_ming_Tan\AppData\Local\anaconda3\envs\fws-dell\Lib\site-packages\django\core\handlers\base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Zhong_ming_Tan\AppData\Local\anaconda3\envs\fws-dell\Lib\site-packages\django\views\generic\base.py", line 104, in view
    return self.dispatch(request, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Zhong_ming_Tan\AppData\Local\anaconda3\envs\fws-dell\Lib\site-packages\django\utils\decorators.py", line 48, in _wrapper
    return bound_method(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Zhong_ming_Tan\AppData\Local\anaconda3\envs\fws-dell\Lib\site-packages\django\views\decorators\debug.py", line 141, in sensitive_post_parameters_wrapper
    return view(request, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Zhong_ming_Tan\AppData\Local\anaconda3\envs\fws-dell\Lib\site-packages\django\utils\decorators.py", line 48, in _wrapper
    return bound_method(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Zhong_ming_Tan\AppData\Local\anaconda3\envs\fws-dell\Lib\site-packages\django\utils\decorators.py", line 188, in _view_wrapper
    result = _process_exception(request, e)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Zhong_ming_Tan\AppData\Local\anaconda3\envs\fws-dell\Lib\site-packages\django\utils\decorators.py", line 186, in _view_wrapper
    response = view_func(request, *args, **kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Zhong_ming_Tan\AppData\Local\anaconda3\envs\fws-dell\Lib\site-packages\django\utils\decorators.py", line 48, in _wrapper
    return bound_method(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Zhong_ming_Tan\AppData\Local\anaconda3\envs\fws-dell\Lib\site-packages\django\views\decorators\cache.py", line 80, in _view_wrapper
    response = view_func(request, *args, **kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Zhong_ming_Tan\AppData\Local\anaconda3\envs\fws-dell\Lib\site-packages\django\contrib\auth\views.py", line 88, in dispatch
    return super().dispatch(request, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Zhong_ming_Tan\AppData\Local\anaconda3\envs\fws-dell\Lib\site-packages\django\views\generic\base.py", line 143, in dispatch
    return handler(request, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Zhong_ming_Tan\AppData\Local\anaconda3\envs\fws-dell\Lib\site-packages\django\views\generic\edit.py", line 151, in post
    return self.form_valid(form)
           ^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Zhong_ming_Tan\AppData\Local\anaconda3\envs\fws-dell\Lib\site-packages\django\contrib\auth\views.py", line 108, in form_valid
    return HttpResponseRedirect(self.get_success_url())
                                ^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Zhong_ming_Tan\AppData\Local\anaconda3\envs\fws-dell\Lib\site-packages\django\contrib\auth\views.py", line 41, in get_success_url
    return self.get_redirect_url() or self.get_default_redirect_url()
                                      ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Zhong_ming_Tan\AppData\Local\anaconda3\envs\fws-dell\Lib\site-packages\django\contrib\auth\views.py", line 95, in get_default_redirect_url
    return resolve_url(settings.LOGIN_REDIRECT_URL)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Zhong_ming_Tan\AppData\Local\anaconda3\envs\fws-dell\Lib\site-packages\django\shortcuts.py", line 180, in resolve_url
    return reverse(to, args=args, kwargs=kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Zhong_ming_Tan\AppData\Local\anaconda3\envs\fws-dell\Lib\site-packages\django\urls\base.py", line 88, in reverse
    return resolver._reverse_with_prefix(view, prefix, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Zhong_ming_Tan\AppData\Local\anaconda3\envs\fws-dell\Lib\site-packages\django\urls\resolvers.py", line 851, in _reverse_with_prefix
    raise NoReverseMatch(msg)
django.urls.exceptions.NoReverseMatch: Reverse for 'scrap' not found. 'scrap' is not a valid view function or pattern name.
2024-02-28 00:02:35,501 - INFO - C:\Other Project\Ads\Bitcoin Scraping\scrapperE\scrapperE\urls.py changed, reloading.
2024-02-28 00:03:58,169 - INFO - C:\Other Project\Ads\Bitcoin Scraping\scrapperE\scrapperE\settings.py changed, reloading.
2024-02-28 00:05:45,868 - WARNING - Method Not Allowed (GET): /logout/
2024-02-28 00:05:45,946 - WARNING - Method Not Allowed: /logout/
2024-02-28 00:05:50,192 - WARNING - Method Not Allowed (GET): /logout/
2024-02-28 00:05:50,193 - WARNING - Method Not Allowed: /logout/
2024-02-28 00:05:54,982 - WARNING - Method Not Allowed (GET): /logout/
2024-02-28 00:05:54,982 - WARNING - Method Not Allowed: /logout/
2024-02-28 00:07:49,702 - INFO - C:\Other Project\Ads\Bitcoin Scraping\scrapperE\scrapperE\urls.py changed, reloading.
2024-02-28 00:07:57,050 - WARNING - Method Not Allowed (GET): /logout/
2024-02-28 00:07:57,052 - WARNING - Method Not Allowed: /logout/
2024-02-28 00:07:58,328 - WARNING - Method Not Allowed (GET): /logout/
2024-02-28 00:07:58,329 - WARNING - Method Not Allowed: /logout/
2024-02-28 00:07:58,497 - WARNING - Method Not Allowed (GET): /logout/
2024-02-28 00:07:58,497 - WARNING - Method Not Allowed: /logout/
2024-02-28 00:07:58,673 - WARNING - Method Not Allowed (GET): /logout/
2024-02-28 00:07:58,674 - WARNING - Method Not Allowed: /logout/
2024-02-28 00:07:59,241 - WARNING - Method Not Allowed (GET): /logout/
2024-02-28 00:07:59,242 - WARNING - Method Not Allowed: /logout/
2024-02-28 00:07:59,416 - WARNING - Method Not Allowed (GET): /logout/
2024-02-28 00:07:59,416 - WARNING - Method Not Allowed: /logout/
2024-02-28 00:08:16,462 - WARNING - Method Not Allowed (GET): /logout/
2024-02-28 00:08:16,493 - WARNING - Method Not Allowed: /logout/
2024-02-28 00:08:25,271 - WARNING - Method Not Allowed (GET): /logout/
2024-02-28 00:08:25,274 - WARNING - Method Not Allowed: /logout/
2024-02-28 00:08:27,132 - WARNING - Method Not Allowed (GET): /logout/
2024-02-28 00:08:27,133 - WARNING - Method Not Allowed: /logout/
2024-02-28 00:12:25,743 - INFO - C:\Other Project\Ads\Bitcoin Scraping\scrapperE\scrapperE\urls.py changed, reloading.
2024-02-28 00:12:35,054 - INFO - C:\Other Project\Ads\Bitcoin Scraping\scrapperE\scrapperE\settings.py changed, reloading.
2024-02-28 00:12:48,500 - WARNING - Method Not Allowed (GET): /logout/
2024-02-28 00:12:48,579 - WARNING - Method Not Allowed: /logout/
2024-02-28 00:12:53,074 - WARNING - Method Not Allowed (GET): /logout/
2024-02-28 00:12:53,076 - WARNING - Method Not Allowed: /logout/
2024-02-28 00:12:57,179 - WARNING - Not Found: /
2024-02-28 00:13:24,018 - WARNING - Method Not Allowed (GET): /logout/
2024-02-28 00:13:24,080 - WARNING - Method Not Allowed: /logout/
2024-02-28 00:13:37,655 - WARNING - Not Found: /
2024-02-28 00:13:48,963 - WARNING - Method Not Allowed (GET): /logout/
2024-02-28 00:13:48,995 - WARNING - Method Not Allowed: /logout/
2024-02-28 00:17:53,721 - INFO - C:\Other Project\Ads\Bitcoin Scraping\scrapperE\user\views.py changed, reloading.
2024-02-28 00:18:02,203 - INFO - C:\Other Project\Ads\Bitcoin Scraping\scrapperE\user\views.py changed, reloading.
2024-02-28 00:18:54,117 - INFO - C:\Other Project\Ads\Bitcoin Scraping\scrapperE\scrapperE\urls.py changed, reloading.
2024-02-28 00:19:58,863 - INFO - C:\Other Project\Ads\Bitcoin Scraping\scrapperE\scrap\views.py changed, reloading.
2024-02-28 00:20:21,238 - INFO - C:\Other Project\Ads\Bitcoin Scraping\scrapperE\scrap\views.py changed, reloading.
2024-02-28 00:20:58,619 - INFO - C:\Other Project\Ads\Bitcoin Scraping\scrapperE\scrap\views.py changed, reloading.
2024-02-28 00:21:19,465 - INFO - C:\Other Project\Ads\Bitcoin Scraping\scrapperE\user\views.py changed, reloading.
2024-02-28 00:21:55,588 - INFO - C:\Other Project\Ads\Bitcoin Scraping\scrapperE\scrapperE\urls.py changed, reloading.
2024-02-28 00:22:14,566 - INFO - C:\Other Project\Ads\Bitcoin Scraping\scrapperE\user\views.py changed, reloading.
2024-02-28 00:22:39,611 - INFO - C:\Other Project\Ads\Bitcoin Scraping\scrapperE\scrapperE\urls.py changed, reloading.
2024-02-28 00:24:11,746 - INFO - C:\Other Project\Ads\Bitcoin Scraping\scrapperE\user\views.py changed, reloading.
2024-02-28 00:24:26,494 - WARNING - Not Found: /logout/user/logout.html
2024-02-28 00:25:00,964 - INFO - C:\Other Project\Ads\Bitcoin Scraping\scrapperE\user\views.py changed, reloading.
